// vue-qrcode-reader/dist/vue-qrcode-reader.js
import { defineComponent as St, ref as Re, onMounted as Zo, onUnmounted as Xo, computed as or, watch as ir, openBlock as Pt, createElementBlock as $t, createElementVNode as Ze, normalizeClass as ei, withDirectives as ti, vShow as ri, renderSlot as dr, withModifiers as Xe } from "vue";
var pr = (n, a, o) => {
  if (!a.has(n))
    throw TypeError("Cannot " + o);
}, ni = (n, a, o) => (pr(n, a, "read from private field"), o ? o.call(n) : a.get(n)), oi = (n, a, o) => {
  if (a.has(n))
    throw TypeError("Cannot add the same private member more than once");
  a instanceof WeakSet ? a.add(n) : a.set(n, o);
}, ii = (n, a, o, u) => (pr(n, a, "write to private field"), u ? u.call(n, o) : a.set(n, o), o);
function ai(n) {
  if (si(n))
    return {
      width: n.naturalWidth,
      height: n.naturalHeight
    };
  if (ci(n))
    return {
      width: n.width.baseVal.value,
      height: n.height.baseVal.value
    };
  if (ui(n))
    return {
      width: n.videoWidth,
      height: n.videoHeight
    };
  if (hr(n))
    return {
      width: n.width,
      height: n.height
    };
  if (fi(n))
    return {
      width: n.width,
      height: n.height
    };
  if (li(n))
    return {
      width: n.width,
      height: n.height
    };
  if (di(n))
    return {
      width: n.displayWidth,
      height: n.displayHeight
    };
  throw new Error("Unsupported image type");
}
function si(n) {
  try {
    return n instanceof HTMLImageElement;
  } catch {
    return !1;
  }
}
function ci(n) {
  try {
    return n instanceof SVGImageElement;
  } catch {
    return !1;
  }
}
function ui(n) {
  try {
    return n instanceof HTMLVideoElement;
  } catch {
    return !1;
  }
}
function hr(n) {
  try {
    return n instanceof HTMLCanvasElement;
  } catch {
    return !1;
  }
}
function fi(n) {
  try {
    return n instanceof ImageBitmap;
  } catch {
    return !1;
  }
}
function li(n) {
  try {
    return n instanceof OffscreenCanvas;
  } catch {
    return !1;
  }
}
function di(n) {
  try {
    return n instanceof VideoFrame;
  } catch {
    return !1;
  }
}
function pi(n) {
  try {
    return n instanceof Blob;
  } catch {
    return !1;
  }
}
function hi(n) {
  try {
    return n instanceof ImageData;
  } catch {
    return !1;
  }
}
function mi(n, a) {
  try {
    const o = new OffscreenCanvas(n, a);
    if (o.getContext("2d") instanceof OffscreenCanvasRenderingContext2D)
      return o;
    throw new Error("OffscreenCanvasRenderingContext2D is not supported");
  } catch {
    const o = document.createElement("canvas");
    return o.width = n, o.height = a, o;
  }
}
function mr(n) {
  const { width: a, height: o } = ai(n), u = mi(a, o), d = u.getContext("2d");
  d.drawImage(n, 0, 0);
  const p = d.getImageData(0, 0, a, o);
  return hr(u) && u.remove(), p;
}
async function vi(n) {
  const a = await createImageBitmap(n);
  return mr(a);
}
async function yi(n) {
  return pi(n) ? await vi(n) : hi(n) ? n : mr(n);
}
const ar = (n) => {
  let a;
  const o = /* @__PURE__ */ new Set(), u = (m, w) => {
    const S = typeof m == "function" ? m(a) : m;
    if (!Object.is(S, a)) {
      const v = a;
      a = w ?? typeof S != "object" ? S : Object.assign({}, a, S), o.forEach((b) => b(a, v));
    }
  }, d = () => a, p = { setState: u, getState: d, subscribe: (m) => (o.add(m), () => o.delete(m)), destroy: () => {
    ({ BASE_URL: "/", MODE: "production", DEV: !1, PROD: !0, SSR: !1 } && "production") !== "production" && console.warn(
      "[DEPRECATED] The `destroy` method will be unsupported in a future version. Instead use unsubscribe function returned by subscribe. Everything will be garbage-collected if store is garbage-collected."
    ), o.clear();
  } };
  return a = n(u, d, p), p;
}, gi = (n) => n ? ar(n) : ar, Ci = {
  locateFile: (n, a) => {
    var o;
    const u = (o = n.match(new RegExp("(?<=_).+?(?=\\.wasm$)"))) == null ? void 0 : o[0];
    return u ? `https://cdn.jsdelivr.net/npm/@sec-ant/zxing-wasm@1.2.4/dist/${u}/${n}` : a + n;
  }
}, xe = gi()(() => ({
  zxingModuleWeakMap: /* @__PURE__ */ new WeakMap(),
  zxingModuleOverrides: Ci
}));
function vr(n, a = xe.getState().zxingModuleOverrides) {
  const { zxingModuleWeakMap: o } = xe.getState(), u = o.get(
    n
  );
  if (u && Object.is(a, xe.getState().zxingModuleOverrides))
    return u;
  {
    xe.setState({
      zxingModuleOverrides: a
    });
    const d = n(a);
    return o.set(n, d), d;
  }
}
const sr = [
  "Aztec",
  "Codabar",
  "Code128",
  "Code39",
  "Code93",
  "DataBar",
  "DataBarExpanded",
  "DataMatrix",
  "EAN-13",
  "EAN-8",
  "ITF",
  "Linear-Codes",
  "Matrix-Codes",
  "MaxiCode",
  "MicroQRCode",
  "None",
  "PDF417",
  "QRCode",
  "UPC-A",
  "UPC-E"
], fe = {
  tryHarder: !0,
  formats: [],
  maxNumberOfSymbols: 1 / 0
};
async function wi(n, {
  tryHarder: a = fe.tryHarder,
  formats: o = fe.formats,
  maxNumberOfSymbols: u = fe.maxNumberOfSymbols
} = fe, d) {
  const p = await vr(
    d,
    xe.getState().zxingModuleOverrides
  ), {
    data: m,
    width: w,
    height: S,
    data: { byteLength: v }
  } = n, b = p._malloc(v);
  p.HEAP8.set(m, b);
  const $ = p.readBarcodeFromPixmap(
    b,
    w,
    S,
    a,
    _i(o),
    u
  );
  p._free(b);
  const C = [];
  for (let E = 0; E < $.size(); ++E) {
    const A = $.get(E);
    C.push({
      ...A,
      format: bi(A.format)
    });
  }
  return C;
}
function _i(n) {
  return n.join("|");
}
function bi(n) {
  const a = cr(n);
  let o = 0, u = sr.length - 1;
  for (; o <= u; ) {
    const d = Math.floor((o + u) / 2), p = sr[d], m = cr(p);
    if (m === a)
      return p;
    m < a ? o = d + 1 : u = d - 1;
  }
  return "None";
}
function cr(n) {
  return n.toLowerCase().replace(/_-\[\]/g, "");
}
var yr = (() => {
  var n = import.meta.url;
  return function(a = {}) {
    var o = typeof a < "u" ? a : {}, u, d;
    o.ready = new Promise((e, t) => {
      u = e, d = t;
    });
    var p = Object.assign({}, o), m = "./this.program", w = typeof window == "object", S = typeof importScripts == "function";
    typeof process == "object" && typeof process.versions == "object" && process.versions.node;
    var v = "";
    function b(e) {
      return o.locateFile ? o.locateFile(e, v) : v + e;
    }
    var $;
    (w || S) && (S ? v = self.location.href : typeof document < "u" && document.currentScript && (v = document.currentScript.src), n && (v = n), v.indexOf("blob:") !== 0 ? v = v.substr(0, v.replace(/[?#].*/, "").lastIndexOf("/") + 1) : v = "", S && ($ = (e) => {
      var t = new XMLHttpRequest();
      return t.open("GET", e, !1), t.responseType = "arraybuffer", t.send(null), new Uint8Array(t.response);
    })), o.print || console.log.bind(console);
    var C = o.printErr || console.error.bind(console);
    Object.assign(o, p), p = null, o.arguments && o.arguments, o.thisProgram && (m = o.thisProgram), o.quit && o.quit;
    var E;
    o.wasmBinary && (E = o.wasmBinary), o.noExitRuntime, typeof WebAssembly != "object" && we("no native wasm support detected");
    var A, F = !1;
    function B(e, t) {
      e || we(t);
    }
    var I, L, ee, pe, H, V, ke, Ue;
    function Fe() {
      var e = A.buffer;
      o.HEAP8 = I = new Int8Array(e), o.HEAP16 = ee = new Int16Array(e), o.HEAP32 = H = new Int32Array(e), o.HEAPU8 = L = new Uint8Array(e), o.HEAPU16 = pe = new Uint16Array(e), o.HEAPU32 = V = new Uint32Array(e), o.HEAPF32 = ke = new Float32Array(e), o.HEAPF64 = Ue = new Float64Array(e);
    }
    var Ce, Pe = [], Ie = [], We = [];
    function se() {
      if (o.preRun)
        for (typeof o.preRun == "function" && (o.preRun = [o.preRun]); o.preRun.length; )
          it(o.preRun.shift());
      ct(Pe);
    }
    function Z() {
      ct(Ie);
    }
    function ot() {
      if (o.postRun)
        for (typeof o.postRun == "function" && (o.postRun = [o.postRun]); o.postRun.length; )
          st(o.postRun.shift());
      ct(We);
    }
    function it(e) {
      Pe.unshift(e);
    }
    function at(e) {
      Ie.unshift(e);
    }
    function st(e) {
      We.unshift(e);
    }
    var te = 0, he = null;
    function Rr(e) {
      te++, o.monitorRunDependencies && o.monitorRunDependencies(te);
    }
    function xr(e) {
      if (te--, o.monitorRunDependencies && o.monitorRunDependencies(te), te == 0 && he) {
        var t = he;
        he = null, t();
      }
    }
    function we(e) {
      o.onAbort && o.onAbort(e), e = "Aborted(" + e + ")", C(e), F = !0, e += ". Build with -sASSERTIONS for more info.";
      var t = new WebAssembly.RuntimeError(e);
      throw d(t), t;
    }
    var kr = "data:application/octet-stream;base64,";
    function Tt(e) {
      return e.startsWith(kr);
    }
    var me;
    o.locateFile ? (me = "zxing_reader.wasm", Tt(me) || (me = b(me))) : me = new URL("/reader/zxing_reader.wasm", self.location).href;
    function Ot(e) {
      try {
        if (e == me && E)
          return new Uint8Array(E);
        if ($)
          return $(e);
        throw "both async and sync fetching of the wasm failed";
      } catch (t) {
        we(t);
      }
    }
    function Ur(e) {
      return !E && (w || S) && typeof fetch == "function" ? fetch(e, { credentials: "same-origin" }).then((t) => {
        if (!t.ok)
          throw "failed to load wasm binary file at '" + e + "'";
        return t.arrayBuffer();
      }).catch(() => Ot(e)) : Promise.resolve().then(() => Ot(e));
    }
    function Et(e, t, r) {
      return Ur(e).then((i) => WebAssembly.instantiate(i, t)).then((i) => i).then(r, (i) => {
        C("failed to asynchronously prepare wasm: " + i), we(i);
      });
    }
    function Fr(e, t, r, i) {
      return !e && typeof WebAssembly.instantiateStreaming == "function" && !Tt(t) && typeof fetch == "function" ? fetch(t, { credentials: "same-origin" }).then((s) => {
        var c = WebAssembly.instantiateStreaming(s, r);
        return c.then(i, function(l) {
          return C("wasm streaming compile failed: " + l), C("falling back to ArrayBuffer instantiation"), Et(t, r, i);
        });
      }) : Et(t, r, i);
    }
    function Ir() {
      var e = { a: Co };
      function t(i, s) {
        var c = i.exports;
        return o.asm = c, A = o.asm.oa, Fe(), Ce = o.asm.sa, at(o.asm.pa), xr(), c;
      }
      Rr();
      function r(i) {
        t(i.instance);
      }
      if (o.instantiateWasm)
        try {
          return o.instantiateWasm(e, t);
        } catch (i) {
          C("Module.instantiateWasm callback failed with error: " + i), d(i);
        }
      return Fr(E, me, e, r).catch(d), {};
    }
    function ct(e) {
      for (; e.length > 0; )
        e.shift()(o);
    }
    var je = [], Le = 0;
    function Wr(e) {
      var t = new He(e);
      return t.get_caught() || (t.set_caught(!0), Le--), t.set_rethrown(!1), je.push(t), Zt(t.excPtr), t.get_exception_ptr();
    }
    var re = 0;
    function jr() {
      D(0);
      var e = je.pop();
      Kt(e.excPtr), re = 0;
    }
    function He(e) {
      this.excPtr = e, this.ptr = e - 24, this.set_type = function(t) {
        V[this.ptr + 4 >> 2] = t;
      }, this.get_type = function() {
        return V[this.ptr + 4 >> 2];
      }, this.set_destructor = function(t) {
        V[this.ptr + 8 >> 2] = t;
      }, this.get_destructor = function() {
        return V[this.ptr + 8 >> 2];
      }, this.set_caught = function(t) {
        t = t ? 1 : 0, I[this.ptr + 12 >> 0] = t;
      }, this.get_caught = function() {
        return I[this.ptr + 12 >> 0] != 0;
      }, this.set_rethrown = function(t) {
        t = t ? 1 : 0, I[this.ptr + 13 >> 0] = t;
      }, this.get_rethrown = function() {
        return I[this.ptr + 13 >> 0] != 0;
      }, this.init = function(t, r) {
        this.set_adjusted_ptr(0), this.set_type(t), this.set_destructor(r);
      }, this.set_adjusted_ptr = function(t) {
        V[this.ptr + 16 >> 2] = t;
      }, this.get_adjusted_ptr = function() {
        return V[this.ptr + 16 >> 2];
      }, this.get_exception_ptr = function() {
        var t = er(this.get_type());
        if (t)
          return V[this.excPtr >> 2];
        var r = this.get_adjusted_ptr();
        return r !== 0 ? r : this.excPtr;
      };
    }
    function Lr(e) {
      throw re || (re = e), re;
    }
    function ut() {
      var e = re;
      if (!e)
        return Me(0), 0;
      var t = new He(e);
      t.set_adjusted_ptr(e);
      var r = t.get_type();
      if (!r)
        return Me(0), e;
      for (var i = 0; i < arguments.length; i++) {
        var s = arguments[i];
        if (s === 0 || s === r)
          break;
        var c = t.ptr + 16;
        if (Xt(s, r, c))
          return Me(s), e;
      }
      return Me(r), e;
    }
    var Hr = ut, Vr = ut, zr = ut;
    function Br(e) {
      var t = new He(e).get_exception_ptr();
      return t;
    }
    function Nr() {
      var e = je.pop();
      e || we("no exception to throw");
      var t = e.excPtr;
      throw e.get_rethrown() || (je.push(e), e.set_rethrown(!0), e.set_caught(!1), Le++), re = t, re;
    }
    function Gr(e, t, r) {
      var i = new He(e);
      throw i.init(t, r), re = e, Le++, re;
    }
    function qr() {
      return Le;
    }
    var Ve = {};
    function At(e) {
      for (; e.length; ) {
        var t = e.pop(), r = e.pop();
        r(t);
      }
    }
    function $e(e) {
      return this.fromWireType(H[e >> 2]);
    }
    var _e = {}, ve = {}, ze = {}, Yr = 48, Jr = 57;
    function Dt(e) {
      if (e === void 0)
        return "_unknown";
      e = e.replace(/[^a-zA-Z0-9_]/g, "$");
      var t = e.charCodeAt(0);
      return t >= Yr && t <= Jr ? `_${e}` : e;
    }
    function Mt(e, t) {
      return e = Dt(e), { [e]: function() {
        return t.apply(this, arguments);
      } }[e];
    }
    function ft(e, t) {
      var r = Mt(t, function(i) {
        this.name = t, this.message = i;
        var s = new Error(i).stack;
        s !== void 0 && (this.stack = this.toString() + `
` + s.replace(/^Error(:[^\n]*)?\n/, ""));
      });
      return r.prototype = Object.create(e.prototype), r.prototype.constructor = r, r.prototype.toString = function() {
        return this.message === void 0 ? this.name : `${this.name}: ${this.message}`;
      }, r;
    }
    var Rt = void 0;
    function Be(e) {
      throw new Rt(e);
    }
    function ye(e, t, r) {
      e.forEach(function(f) {
        ze[f] = t;
      });
      function i(f) {
        var h = r(f);
        h.length !== e.length && Be("Mismatched type converter count");
        for (var y = 0; y < e.length; ++y)
          ne(e[y], h[y]);
      }
      var s = new Array(t.length), c = [], l = 0;
      t.forEach((f, h) => {
        ve.hasOwnProperty(f) ? s[h] = ve[f] : (c.push(f), _e.hasOwnProperty(f) || (_e[f] = []), _e[f].push(() => {
          s[h] = ve[f], ++l, l === c.length && i(s);
        }));
      }), c.length === 0 && i(s);
    }
    function Qr(e) {
      var t = Ve[e];
      delete Ve[e];
      var r = t.rawConstructor, i = t.rawDestructor, s = t.fields, c = s.map((l) => l.getterReturnType).concat(s.map((l) => l.setterArgumentType));
      ye([e], c, (l) => {
        var f = {};
        return s.forEach((h, y) => {
          var _ = h.fieldName, T = l[y], O = h.getter, x = h.getterContext, W = l[y + s.length], G = h.setter, j = h.setterContext;
          f[_] = { read: (q) => T.fromWireType(O(x, q)), write: (q, g) => {
            var P = [];
            G(j, q, W.toWireType(P, g)), At(P);
          } };
        }), [{ name: t.name, fromWireType: function(h) {
          var y = {};
          for (var _ in f)
            y[_] = f[_].read(h);
          return i(h), y;
        }, toWireType: function(h, y) {
          for (var _ in f)
            if (!(_ in y))
              throw new TypeError(`Missing field: "${_}"`);
          var T = r();
          for (_ in f)
            f[_].write(T, y[_]);
          return h !== null && h.push(i, T), T;
        }, argPackAdvance: 8, readValueFromPointer: $e, destructorFunction: i }];
      });
    }
    function Kr(e, t, r, i, s) {
    }
    function lt(e) {
      switch (e) {
        case 1:
          return 0;
        case 2:
          return 1;
        case 4:
          return 2;
        case 8:
          return 3;
        default:
          throw new TypeError(`Unknown type size: ${e}`);
      }
    }
    function Zr() {
      for (var e = new Array(256), t = 0; t < 256; ++t)
        e[t] = String.fromCharCode(t);
      xt = e;
    }
    var xt = void 0;
    function Y(e) {
      for (var t = "", r = e; L[r]; )
        t += xt[L[r++]];
      return t;
    }
    var be = void 0;
    function k(e) {
      throw new be(e);
    }
    function ne(e, t, r = {}) {
      if (!("argPackAdvance" in t))
        throw new TypeError("registerType registeredInstance requires argPackAdvance");
      var i = t.name;
      if (e || k(`type "${i}" must have a positive integer typeid pointer`), ve.hasOwnProperty(e)) {
        if (r.ignoreDuplicateRegistrations)
          return;
        k(`Cannot register type '${i}' twice`);
      }
      if (ve[e] = t, delete ze[e], _e.hasOwnProperty(e)) {
        var s = _e[e];
        delete _e[e], s.forEach((c) => c());
      }
    }
    function Xr(e, t, r, i, s) {
      var c = lt(r);
      t = Y(t), ne(e, { name: t, fromWireType: function(l) {
        return !!l;
      }, toWireType: function(l, f) {
        return f ? i : s;
      }, argPackAdvance: 8, readValueFromPointer: function(l) {
        var f;
        if (r === 1)
          f = I;
        else if (r === 2)
          f = ee;
        else if (r === 4)
          f = H;
        else
          throw new TypeError("Unknown boolean type size: " + t);
        return this.fromWireType(f[l >> c]);
      }, destructorFunction: null });
    }
    function en(e) {
      if (!(this instanceof ce) || !(e instanceof ce))
        return !1;
      for (var t = this.$$.ptrType.registeredClass, r = this.$$.ptr, i = e.$$.ptrType.registeredClass, s = e.$$.ptr; t.baseClass; )
        r = t.upcast(r), t = t.baseClass;
      for (; i.baseClass; )
        s = i.upcast(s), i = i.baseClass;
      return t === i && r === s;
    }
    function tn(e) {
      return { count: e.count, deleteScheduled: e.deleteScheduled, preservePointerOnDelete: e.preservePointerOnDelete, ptr: e.ptr, ptrType: e.ptrType, smartPtr: e.smartPtr, smartPtrType: e.smartPtrType };
    }
    function dt(e) {
      function t(r) {
        return r.$$.ptrType.registeredClass.name;
      }
      k(t(e) + " instance already deleted");
    }
    var pt = !1;
    function kt(e) {
    }
    function rn(e) {
      e.smartPtr ? e.smartPtrType.rawDestructor(e.smartPtr) : e.ptrType.registeredClass.rawDestructor(e.ptr);
    }
    function Ut(e) {
      e.count.value -= 1;
      var t = e.count.value === 0;
      t && rn(e);
    }
    function Ft(e, t, r) {
      if (t === r)
        return e;
      if (r.baseClass === void 0)
        return null;
      var i = Ft(e, t, r.baseClass);
      return i === null ? null : r.downcast(i);
    }
    var It = {};
    function nn() {
      return Object.keys(Ee).length;
    }
    function on() {
      var e = [];
      for (var t in Ee)
        Ee.hasOwnProperty(t) && e.push(Ee[t]);
      return e;
    }
    var Te = [];
    function ht() {
      for (; Te.length; ) {
        var e = Te.pop();
        e.$$.deleteScheduled = !1, e.delete();
      }
    }
    var Oe = void 0;
    function an(e) {
      Oe = e, Te.length && Oe && Oe(ht);
    }
    function sn() {
      o.getInheritedInstanceCount = nn, o.getLiveInheritedInstances = on, o.flushPendingDeletes = ht, o.setDelayFunction = an;
    }
    var Ee = {};
    function cn(e, t) {
      for (t === void 0 && k("ptr should not be undefined"); e.baseClass; )
        t = e.upcast(t), e = e.baseClass;
      return t;
    }
    function un(e, t) {
      return t = cn(e, t), Ee[t];
    }
    function Ne(e, t) {
      (!t.ptrType || !t.ptr) && Be("makeClassHandle requires ptr and ptrType");
      var r = !!t.smartPtrType, i = !!t.smartPtr;
      return r !== i && Be("Both smartPtrType and smartPtr must be specified"), t.count = { value: 1 }, Ae(Object.create(e, { $$: { value: t } }));
    }
    function fn(e) {
      var t = this.getPointee(e);
      if (!t)
        return this.destructor(e), null;
      var r = un(this.registeredClass, t);
      if (r !== void 0) {
        if (r.$$.count.value === 0)
          return r.$$.ptr = t, r.$$.smartPtr = e, r.clone();
        var i = r.clone();
        return this.destructor(e), i;
      }
      function s() {
        return this.isSmartPointer ? Ne(this.registeredClass.instancePrototype, { ptrType: this.pointeeType, ptr: t, smartPtrType: this, smartPtr: e }) : Ne(this.registeredClass.instancePrototype, { ptrType: this, ptr: e });
      }
      var c = this.registeredClass.getActualType(t), l = It[c];
      if (!l)
        return s.call(this);
      var f;
      this.isConst ? f = l.constPointerType : f = l.pointerType;
      var h = Ft(t, this.registeredClass, f.registeredClass);
      return h === null ? s.call(this) : this.isSmartPointer ? Ne(f.registeredClass.instancePrototype, { ptrType: f, ptr: h, smartPtrType: this, smartPtr: e }) : Ne(f.registeredClass.instancePrototype, { ptrType: f, ptr: h });
    }
    function Ae(e) {
      return typeof FinalizationRegistry > "u" ? (Ae = (t) => t, e) : (pt = new FinalizationRegistry((t) => {
        Ut(t.$$);
      }), Ae = (t) => {
        var r = t.$$, i = !!r.smartPtr;
        if (i) {
          var s = { $$: r };
          pt.register(t, s, t);
        }
        return t;
      }, kt = (t) => pt.unregister(t), Ae(e));
    }
    function ln() {
      if (this.$$.ptr || dt(this), this.$$.preservePointerOnDelete)
        return this.$$.count.value += 1, this;
      var e = Ae(Object.create(Object.getPrototypeOf(this), { $$: { value: tn(this.$$) } }));
      return e.$$.count.value += 1, e.$$.deleteScheduled = !1, e;
    }
    function dn() {
      this.$$.ptr || dt(this), this.$$.deleteScheduled && !this.$$.preservePointerOnDelete && k("Object already scheduled for deletion"), kt(this), Ut(this.$$), this.$$.preservePointerOnDelete || (this.$$.smartPtr = void 0, this.$$.ptr = void 0);
    }
    function pn() {
      return !this.$$.ptr;
    }
    function hn() {
      return this.$$.ptr || dt(this), this.$$.deleteScheduled && !this.$$.preservePointerOnDelete && k("Object already scheduled for deletion"), Te.push(this), Te.length === 1 && Oe && Oe(ht), this.$$.deleteScheduled = !0, this;
    }
    function mn() {
      ce.prototype.isAliasOf = en, ce.prototype.clone = ln, ce.prototype.delete = dn, ce.prototype.isDeleted = pn, ce.prototype.deleteLater = hn;
    }
    function ce() {
    }
    function Wt(e, t, r) {
      if (e[t].overloadTable === void 0) {
        var i = e[t];
        e[t] = function() {
          return e[t].overloadTable.hasOwnProperty(arguments.length) || k(`Function '${r}' called with an invalid number of arguments (${arguments.length}) - expects one of (${e[t].overloadTable})!`), e[t].overloadTable[arguments.length].apply(this, arguments);
        }, e[t].overloadTable = [], e[t].overloadTable[i.argCount] = i;
      }
    }
    function jt(e, t, r) {
      o.hasOwnProperty(e) ? ((r === void 0 || o[e].overloadTable !== void 0 && o[e].overloadTable[r] !== void 0) && k(`Cannot register public name '${e}' twice`), Wt(o, e, e), o.hasOwnProperty(r) && k(`Cannot register multiple overloads of a function with the same number of arguments (${r})!`), o[e].overloadTable[r] = t) : (o[e] = t, r !== void 0 && (o[e].numArguments = r));
    }
    function vn(e, t, r, i, s, c, l, f) {
      this.name = e, this.constructor = t, this.instancePrototype = r, this.rawDestructor = i, this.baseClass = s, this.getActualType = c, this.upcast = l, this.downcast = f, this.pureVirtualFunctions = [];
    }
    function mt(e, t, r) {
      for (; t !== r; )
        t.upcast || k(`Expected null or instance of ${r.name}, got an instance of ${t.name}`), e = t.upcast(e), t = t.baseClass;
      return e;
    }
    function yn(e, t) {
      if (t === null)
        return this.isReference && k(`null is not a valid ${this.name}`), 0;
      t.$$ || k(`Cannot pass "${gt(t)}" as a ${this.name}`), t.$$.ptr || k(`Cannot pass deleted object as a pointer of type ${this.name}`);
      var r = t.$$.ptrType.registeredClass, i = mt(t.$$.ptr, r, this.registeredClass);
      return i;
    }
    function gn(e, t) {
      var r;
      if (t === null)
        return this.isReference && k(`null is not a valid ${this.name}`), this.isSmartPointer ? (r = this.rawConstructor(), e !== null && e.push(this.rawDestructor, r), r) : 0;
      t.$$ || k(`Cannot pass "${gt(t)}" as a ${this.name}`), t.$$.ptr || k(`Cannot pass deleted object as a pointer of type ${this.name}`), !this.isConst && t.$$.ptrType.isConst && k(`Cannot convert argument of type ${t.$$.smartPtrType ? t.$$.smartPtrType.name : t.$$.ptrType.name} to parameter type ${this.name}`);
      var i = t.$$.ptrType.registeredClass;
      if (r = mt(t.$$.ptr, i, this.registeredClass), this.isSmartPointer)
        switch (t.$$.smartPtr === void 0 && k("Passing raw pointer to smart pointer is illegal"), this.sharingPolicy) {
          case 0:
            t.$$.smartPtrType === this ? r = t.$$.smartPtr : k(`Cannot convert argument of type ${t.$$.smartPtrType ? t.$$.smartPtrType.name : t.$$.ptrType.name} to parameter type ${this.name}`);
            break;
          case 1:
            r = t.$$.smartPtr;
            break;
          case 2:
            if (t.$$.smartPtrType === this)
              r = t.$$.smartPtr;
            else {
              var s = t.clone();
              r = this.rawShare(r, Ye.toHandle(function() {
                s.delete();
              })), e !== null && e.push(this.rawDestructor, r);
            }
            break;
          default:
            k("Unsupporting sharing policy");
        }
      return r;
    }
    function Cn(e, t) {
      if (t === null)
        return this.isReference && k(`null is not a valid ${this.name}`), 0;
      t.$$ || k(`Cannot pass "${gt(t)}" as a ${this.name}`), t.$$.ptr || k(`Cannot pass deleted object as a pointer of type ${this.name}`), t.$$.ptrType.isConst && k(`Cannot convert argument of type ${t.$$.ptrType.name} to parameter type ${this.name}`);
      var r = t.$$.ptrType.registeredClass, i = mt(t.$$.ptr, r, this.registeredClass);
      return i;
    }
    function wn(e) {
      return this.rawGetPointee && (e = this.rawGetPointee(e)), e;
    }
    function _n(e) {
      this.rawDestructor && this.rawDestructor(e);
    }
    function bn(e) {
      e !== null && e.delete();
    }
    function Sn() {
      oe.prototype.getPointee = wn, oe.prototype.destructor = _n, oe.prototype.argPackAdvance = 8, oe.prototype.readValueFromPointer = $e, oe.prototype.deleteObject = bn, oe.prototype.fromWireType = fn;
    }
    function oe(e, t, r, i, s, c, l, f, h, y, _) {
      this.name = e, this.registeredClass = t, this.isReference = r, this.isConst = i, this.isSmartPointer = s, this.pointeeType = c, this.sharingPolicy = l, this.rawGetPointee = f, this.rawConstructor = h, this.rawShare = y, this.rawDestructor = _, !s && t.baseClass === void 0 ? i ? (this.toWireType = yn, this.destructorFunction = null) : (this.toWireType = Cn, this.destructorFunction = null) : this.toWireType = gn;
    }
    function Lt(e, t, r) {
      o.hasOwnProperty(e) || Be("Replacing nonexistant public symbol"), o[e].overloadTable !== void 0 && r !== void 0 ? o[e].overloadTable[r] = t : (o[e] = t, o[e].argCount = r);
    }
    function Pn(e, t, r) {
      var i = o["dynCall_" + e];
      return r && r.length ? i.apply(null, [t].concat(r)) : i.call(null, t);
    }
    var Ge = [];
    function U(e) {
      var t = Ge[e];
      return t || (e >= Ge.length && (Ge.length = e + 1), Ge[e] = t = Ce.get(e)), t;
    }
    function $n(e, t, r) {
      if (e.includes("j"))
        return Pn(e, t, r);
      var i = U(t).apply(null, r);
      return i;
    }
    function Tn(e, t) {
      var r = [];
      return function() {
        return r.length = 0, Object.assign(r, arguments), $n(e, t, r);
      };
    }
    function K(e, t) {
      e = Y(e);
      function r() {
        return e.includes("j") ? Tn(e, t) : U(t);
      }
      var i = r();
      return typeof i != "function" && k(`unknown function pointer with signature ${e}: ${t}`), i;
    }
    var Ht = void 0;
    function Vt(e) {
      var t = Qt(e), r = Y(t);
      return ie(t), r;
    }
    function qe(e, t) {
      var r = [], i = {};
      function s(c) {
        if (!i[c] && !ve[c]) {
          if (ze[c]) {
            ze[c].forEach(s);
            return;
          }
          r.push(c), i[c] = !0;
        }
      }
      throw t.forEach(s), new Ht(`${e}: ` + r.map(Vt).join([", "]));
    }
    function On(e, t, r, i, s, c, l, f, h, y, _, T, O) {
      _ = Y(_), c = K(s, c), f && (f = K(l, f)), y && (y = K(h, y)), O = K(T, O);
      var x = Dt(_);
      jt(x, function() {
        qe(`Cannot construct ${_} due to unbound types`, [i]);
      }), ye([e, t, r], i ? [i] : [], function(W) {
        W = W[0];
        var G, j;
        i ? (G = W.registeredClass, j = G.instancePrototype) : j = ce.prototype;
        var q = Mt(x, function() {
          if (Object.getPrototypeOf(this) !== g)
            throw new be("Use 'new' to construct " + _);
          if (P.constructor_body === void 0)
            throw new be(_ + " has no accessible constructor");
          var Ke = P.constructor_body[arguments.length];
          if (Ke === void 0)
            throw new be(`Tried to invoke ctor of ${_} with invalid number of parameters (${arguments.length}) - expected (${Object.keys(P.constructor_body).toString()}) parameters instead!`);
          return Ke.apply(this, arguments);
        }), g = Object.create(j, { constructor: { value: q } });
        q.prototype = g;
        var P = new vn(_, q, g, O, G, c, f, y);
        P.baseClass && (P.baseClass.__derivedClasses === void 0 && (P.baseClass.__derivedClasses = []), P.baseClass.__derivedClasses.push(P));
        var z = new oe(_, P, !0, !1, !1), N = new oe(_ + "*", P, !1, !1, !1), ge = new oe(_ + " const*", P, !1, !0, !1);
        return It[e] = { pointerType: N, constPointerType: ge }, Lt(x, q), [z, N, ge];
      });
    }
    function vt(e, t) {
      for (var r = [], i = 0; i < e; i++)
        r.push(V[t + i * 4 >> 2]);
      return r;
    }
    function yt(e, t, r, i, s, c) {
      var l = t.length;
      l < 2 && k("argTypes array size mismatch! Must at least get return value and 'this' types!");
      for (var f = t[1] !== null && r !== null, h = !1, y = 1; y < t.length; ++y)
        if (t[y] !== null && t[y].destructorFunction === void 0) {
          h = !0;
          break;
        }
      var _ = t[0].name !== "void", T = l - 2, O = new Array(T), x = [], W = [];
      return function() {
        arguments.length !== T && k(`function ${e} called with ${arguments.length} arguments, expected ${T} args!`), W.length = 0;
        var G;
        x.length = f ? 2 : 1, x[0] = s, f && (G = t[1].toWireType(W, this), x[1] = G);
        for (var j = 0; j < T; ++j)
          O[j] = t[j + 2].toWireType(W, arguments[j]), x.push(O[j]);
        var q = i.apply(null, x);
        function g(P) {
          if (h)
            At(W);
          else
            for (var z = f ? 1 : 2; z < t.length; z++) {
              var N = z === 1 ? G : O[z - 2];
              t[z].destructorFunction !== null && t[z].destructorFunction(N);
            }
          if (_)
            return t[0].fromWireType(P);
        }
        return g(q);
      };
    }
    function En(e, t, r, i, s, c) {
      B(t > 0);
      var l = vt(t, r);
      s = K(i, s), ye([], [e], function(f) {
        f = f[0];
        var h = `constructor ${f.name}`;
        if (f.registeredClass.constructor_body === void 0 && (f.registeredClass.constructor_body = []), f.registeredClass.constructor_body[t - 1] !== void 0)
          throw new be(`Cannot register multiple constructors with identical number of parameters (${t - 1}) for class '${f.name}'! Overload resolution is currently only performed using the parameter count, not actual type info!`);
        return f.registeredClass.constructor_body[t - 1] = () => {
          qe(`Cannot construct ${f.name} due to unbound types`, l);
        }, ye([], l, function(y) {
          return y.splice(1, 0, null), f.registeredClass.constructor_body[t - 1] = yt(h, y, null, s, c), [];
        }), [];
      });
    }
    function An(e, t, r, i, s, c, l, f, h) {
      var y = vt(r, i);
      t = Y(t), c = K(s, c), ye([], [e], function(_) {
        _ = _[0];
        var T = `${_.name}.${t}`;
        t.startsWith("@@") && (t = Symbol[t.substring(2)]), f && _.registeredClass.pureVirtualFunctions.push(t);
        function O() {
          qe(`Cannot call ${T} due to unbound types`, y);
        }
        var x = _.registeredClass.instancePrototype, W = x[t];
        return W === void 0 || W.overloadTable === void 0 && W.className !== _.name && W.argCount === r - 2 ? (O.argCount = r - 2, O.className = _.name, x[t] = O) : (Wt(x, t, T), x[t].overloadTable[r - 2] = O), ye([], y, function(G) {
          var j = yt(T, G, _, c, l);
          return x[t].overloadTable === void 0 ? (j.argCount = r - 2, x[t] = j) : x[t].overloadTable[r - 2] = j, [];
        }), [];
      });
    }
    function Dn() {
      this.allocated = [void 0], this.freelist = [], this.get = function(e) {
        return this.allocated[e];
      }, this.has = function(e) {
        return this.allocated[e] !== void 0;
      }, this.allocate = function(e) {
        var t = this.freelist.pop() || this.allocated.length;
        return this.allocated[t] = e, t;
      }, this.free = function(e) {
        this.allocated[e] = void 0, this.freelist.push(e);
      };
    }
    var Q = new Dn();
    function zt(e) {
      e >= Q.reserved && --Q.get(e).refcount === 0 && Q.free(e);
    }
    function Mn() {
      for (var e = 0, t = Q.reserved; t < Q.allocated.length; ++t)
        Q.allocated[t] !== void 0 && ++e;
      return e;
    }
    function Rn() {
      Q.allocated.push({ value: void 0 }, { value: null }, { value: !0 }, { value: !1 }), Q.reserved = Q.allocated.length, o.count_emval_handles = Mn;
    }
    var Ye = { toValue: (e) => (e || k("Cannot use deleted val. handle = " + e), Q.get(e).value), toHandle: (e) => {
      switch (e) {
        case void 0:
          return 1;
        case null:
          return 2;
        case !0:
          return 3;
        case !1:
          return 4;
        default:
          return Q.allocate({ refcount: 1, value: e });
      }
    } };
    function xn(e, t) {
      t = Y(t), ne(e, { name: t, fromWireType: function(r) {
        var i = Ye.toValue(r);
        return zt(r), i;
      }, toWireType: function(r, i) {
        return Ye.toHandle(i);
      }, argPackAdvance: 8, readValueFromPointer: $e, destructorFunction: null });
    }
    function gt(e) {
      if (e === null)
        return "null";
      var t = typeof e;
      return t === "object" || t === "array" || t === "function" ? e.toString() : "" + e;
    }
    function kn(e, t) {
      switch (t) {
        case 2:
          return function(r) {
            return this.fromWireType(ke[r >> 2]);
          };
        case 3:
          return function(r) {
            return this.fromWireType(Ue[r >> 3]);
          };
        default:
          throw new TypeError("Unknown float type: " + e);
      }
    }
    function Un(e, t, r) {
      var i = lt(r);
      t = Y(t), ne(e, { name: t, fromWireType: function(s) {
        return s;
      }, toWireType: function(s, c) {
        return c;
      }, argPackAdvance: 8, readValueFromPointer: kn(t, i), destructorFunction: null });
    }
    function Fn(e, t, r, i, s, c, l) {
      var f = vt(t, r);
      e = Y(e), s = K(i, s), jt(e, function() {
        qe(`Cannot call ${e} due to unbound types`, f);
      }, t - 1), ye([], f, function(h) {
        var y = [h[0], null].concat(h.slice(1));
        return Lt(e, yt(e, y, null, s, c), t - 1), [];
      });
    }
    function In(e, t, r) {
      switch (t) {
        case 0:
          return r ? function(i) {
            return I[i];
          } : function(i) {
            return L[i];
          };
        case 1:
          return r ? function(i) {
            return ee[i >> 1];
          } : function(i) {
            return pe[i >> 1];
          };
        case 2:
          return r ? function(i) {
            return H[i >> 2];
          } : function(i) {
            return V[i >> 2];
          };
        default:
          throw new TypeError("Unknown integer type: " + e);
      }
    }
    function Wn(e, t, r, i, s) {
      t = Y(t);
      var c = lt(r), l = (T) => T;
      if (i === 0) {
        var f = 32 - 8 * r;
        l = (T) => T << f >>> f;
      }
      var h = t.includes("unsigned"), y = (T, O) => {
      }, _;
      h ? _ = function(T, O) {
        return y(O, this.name), O >>> 0;
      } : _ = function(T, O) {
        return y(O, this.name), O;
      }, ne(e, { name: t, fromWireType: l, toWireType: _, argPackAdvance: 8, readValueFromPointer: In(t, c, i !== 0), destructorFunction: null });
    }
    function jn(e, t, r) {
      var i = [Int8Array, Uint8Array, Int16Array, Uint16Array, Int32Array, Uint32Array, Float32Array, Float64Array], s = i[t];
      function c(l) {
        l = l >> 2;
        var f = V, h = f[l], y = f[l + 1];
        return new s(f.buffer, y, h);
      }
      r = Y(r), ne(e, { name: r, fromWireType: c, argPackAdvance: 8, readValueFromPointer: c }, { ignoreDuplicateRegistrations: !0 });
    }
    function Bt(e, t, r, i) {
      if (!(i > 0))
        return 0;
      for (var s = r, c = r + i - 1, l = 0; l < e.length; ++l) {
        var f = e.charCodeAt(l);
        if (f >= 55296 && f <= 57343) {
          var h = e.charCodeAt(++l);
          f = 65536 + ((f & 1023) << 10) | h & 1023;
        }
        if (f <= 127) {
          if (r >= c)
            break;
          t[r++] = f;
        } else if (f <= 2047) {
          if (r + 1 >= c)
            break;
          t[r++] = 192 | f >> 6, t[r++] = 128 | f & 63;
        } else if (f <= 65535) {
          if (r + 2 >= c)
            break;
          t[r++] = 224 | f >> 12, t[r++] = 128 | f >> 6 & 63, t[r++] = 128 | f & 63;
        } else {
          if (r + 3 >= c)
            break;
          t[r++] = 240 | f >> 18, t[r++] = 128 | f >> 12 & 63, t[r++] = 128 | f >> 6 & 63, t[r++] = 128 | f & 63;
        }
      }
      return t[r] = 0, r - s;
    }
    function Ln(e, t, r) {
      return Bt(e, L, t, r);
    }
    function Nt(e) {
      for (var t = 0, r = 0; r < e.length; ++r) {
        var i = e.charCodeAt(r);
        i <= 127 ? t++ : i <= 2047 ? t += 2 : i >= 55296 && i <= 57343 ? (t += 4, ++r) : t += 3;
      }
      return t;
    }
    var Gt = typeof TextDecoder < "u" ? new TextDecoder("utf8") : void 0;
    function Hn(e, t, r) {
      for (var i = t + r, s = t; e[s] && !(s >= i); )
        ++s;
      if (s - t > 16 && e.buffer && Gt)
        return Gt.decode(e.subarray(t, s));
      for (var c = ""; t < s; ) {
        var l = e[t++];
        if (!(l & 128)) {
          c += String.fromCharCode(l);
          continue;
        }
        var f = e[t++] & 63;
        if ((l & 224) == 192) {
          c += String.fromCharCode((l & 31) << 6 | f);
          continue;
        }
        var h = e[t++] & 63;
        if ((l & 240) == 224 ? l = (l & 15) << 12 | f << 6 | h : l = (l & 7) << 18 | f << 12 | h << 6 | e[t++] & 63, l < 65536)
          c += String.fromCharCode(l);
        else {
          var y = l - 65536;
          c += String.fromCharCode(55296 | y >> 10, 56320 | y & 1023);
        }
      }
      return c;
    }
    function Ct(e, t) {
      return e ? Hn(L, e, t) : "";
    }
    function Vn(e, t) {
      t = Y(t);
      var r = t === "std::string";
      ne(e, { name: t, fromWireType: function(i) {
        var s = V[i >> 2], c = i + 4, l;
        if (r)
          for (var f = c, h = 0; h <= s; ++h) {
            var y = c + h;
            if (h == s || L[y] == 0) {
              var _ = y - f, T = Ct(f, _);
              l === void 0 ? l = T : (l += String.fromCharCode(0), l += T), f = y + 1;
            }
          }
        else {
          for (var O = new Array(s), h = 0; h < s; ++h)
            O[h] = String.fromCharCode(L[c + h]);
          l = O.join("");
        }
        return ie(i), l;
      }, toWireType: function(i, s) {
        s instanceof ArrayBuffer && (s = new Uint8Array(s));
        var c, l = typeof s == "string";
        l || s instanceof Uint8Array || s instanceof Uint8ClampedArray || s instanceof Int8Array || k("Cannot pass non-string to std::string"), r && l ? c = Nt(s) : c = s.length;
        var f = _t(4 + c + 1), h = f + 4;
        if (V[f >> 2] = c, r && l)
          Ln(s, h, c + 1);
        else if (l)
          for (var y = 0; y < c; ++y) {
            var _ = s.charCodeAt(y);
            _ > 255 && (ie(h), k("String has UTF-16 code units that do not fit in 8 bits")), L[h + y] = _;
          }
        else
          for (var y = 0; y < c; ++y)
            L[h + y] = s[y];
        return i !== null && i.push(ie, f), f;
      }, argPackAdvance: 8, readValueFromPointer: $e, destructorFunction: function(i) {
        ie(i);
      } });
    }
    var qt = typeof TextDecoder < "u" ? new TextDecoder("utf-16le") : void 0;
    function zn(e, t) {
      for (var r = e, i = r >> 1, s = i + t / 2; !(i >= s) && pe[i]; )
        ++i;
      if (r = i << 1, r - e > 32 && qt)
        return qt.decode(L.subarray(e, r));
      for (var c = "", l = 0; !(l >= t / 2); ++l) {
        var f = ee[e + l * 2 >> 1];
        if (f == 0)
          break;
        c += String.fromCharCode(f);
      }
      return c;
    }
    function Bn(e, t, r) {
      if (r === void 0 && (r = 2147483647), r < 2)
        return 0;
      r -= 2;
      for (var i = t, s = r < e.length * 2 ? r / 2 : e.length, c = 0; c < s; ++c) {
        var l = e.charCodeAt(c);
        ee[t >> 1] = l, t += 2;
      }
      return ee[t >> 1] = 0, t - i;
    }
    function Nn(e) {
      return e.length * 2;
    }
    function Gn(e, t) {
      for (var r = 0, i = ""; !(r >= t / 4); ) {
        var s = H[e + r * 4 >> 2];
        if (s == 0)
          break;
        if (++r, s >= 65536) {
          var c = s - 65536;
          i += String.fromCharCode(55296 | c >> 10, 56320 | c & 1023);
        } else
          i += String.fromCharCode(s);
      }
      return i;
    }
    function qn(e, t, r) {
      if (r === void 0 && (r = 2147483647), r < 4)
        return 0;
      for (var i = t, s = i + r - 4, c = 0; c < e.length; ++c) {
        var l = e.charCodeAt(c);
        if (l >= 55296 && l <= 57343) {
          var f = e.charCodeAt(++c);
          l = 65536 + ((l & 1023) << 10) | f & 1023;
        }
        if (H[t >> 2] = l, t += 4, t + 4 > s)
          break;
      }
      return H[t >> 2] = 0, t - i;
    }
    function Yn(e) {
      for (var t = 0, r = 0; r < e.length; ++r) {
        var i = e.charCodeAt(r);
        i >= 55296 && i <= 57343 && ++r, t += 4;
      }
      return t;
    }
    function Jn(e, t, r) {
      r = Y(r);
      var i, s, c, l, f;
      t === 2 ? (i = zn, s = Bn, l = Nn, c = () => pe, f = 1) : t === 4 && (i = Gn, s = qn, l = Yn, c = () => V, f = 2), ne(e, { name: r, fromWireType: function(h) {
        for (var y = V[h >> 2], _ = c(), T, O = h + 4, x = 0; x <= y; ++x) {
          var W = h + 4 + x * t;
          if (x == y || _[W >> f] == 0) {
            var G = W - O, j = i(O, G);
            T === void 0 ? T = j : (T += String.fromCharCode(0), T += j), O = W + t;
          }
        }
        return ie(h), T;
      }, toWireType: function(h, y) {
        typeof y != "string" && k(`Cannot pass non-string to C++ string type ${r}`);
        var _ = l(y), T = _t(4 + _ + t);
        return V[T >> 2] = _ >> f, s(y, T + 4, _ + t), h !== null && h.push(ie, T), T;
      }, argPackAdvance: 8, readValueFromPointer: $e, destructorFunction: function(h) {
        ie(h);
      } });
    }
    function Qn(e, t, r, i, s, c) {
      Ve[e] = { name: Y(t), rawConstructor: K(r, i), rawDestructor: K(s, c), fields: [] };
    }
    function Kn(e, t, r, i, s, c, l, f, h, y) {
      Ve[e].fields.push({ fieldName: Y(t), getterReturnType: r, getter: K(i, s), getterContext: c, setterArgumentType: l, setter: K(f, h), setterContext: y });
    }
    function Zn(e, t) {
      t = Y(t), ne(e, { isVoid: !0, name: t, argPackAdvance: 0, fromWireType: function() {
      }, toWireType: function(r, i) {
      } });
    }
    function Xn(e) {
      e > 4 && (Q.get(e).refcount += 1);
    }
    function eo(e, t) {
      var r = ve[e];
      return r === void 0 && k(t + " has unknown type " + Vt(e)), r;
    }
    function to(e, t) {
      e = eo(e, "_emval_take_value");
      var r = e.readValueFromPointer(t);
      return Ye.toHandle(r);
    }
    function ro() {
      we("");
    }
    function no(e, t, r) {
      L.copyWithin(e, t, t + r);
    }
    function oo() {
      return 2147483648;
    }
    function io(e) {
      var t = A.buffer, r = e - t.byteLength + 65535 >>> 16;
      try {
        return A.grow(r), Fe(), 1;
      } catch {
      }
    }
    function ao(e) {
      var t = L.length;
      e = e >>> 0;
      var r = oo();
      if (e > r)
        return !1;
      for (var i = (h, y) => h + (y - h % y) % y, s = 1; s <= 4; s *= 2) {
        var c = t * (1 + 0.2 / s);
        c = Math.min(c, e + 100663296);
        var l = Math.min(r, i(Math.max(e, c), 65536)), f = io(l);
        if (f)
          return !0;
      }
      return !1;
    }
    var wt = {};
    function so() {
      return m || "./this.program";
    }
    function De() {
      if (!De.strings) {
        var e = (typeof navigator == "object" && navigator.languages && navigator.languages[0] || "C").replace("-", "_") + ".UTF-8", t = { USER: "web_user", LOGNAME: "web_user", PATH: "/", PWD: "/", HOME: "/home/web_user", LANG: e, _: so() };
        for (var r in wt)
          wt[r] === void 0 ? delete t[r] : t[r] = wt[r];
        var i = [];
        for (var r in t)
          i.push(`${r}=${t[r]}`);
        De.strings = i;
      }
      return De.strings;
    }
    function co(e, t) {
      for (var r = 0; r < e.length; ++r)
        I[t++ >> 0] = e.charCodeAt(r);
      I[t >> 0] = 0;
    }
    function uo(e, t) {
      var r = 0;
      return De().forEach(function(i, s) {
        var c = t + r;
        V[e + s * 4 >> 2] = c, co(i, c), r += i.length + 1;
      }), 0;
    }
    function fo(e, t) {
      var r = De();
      V[e >> 2] = r.length;
      var i = 0;
      return r.forEach(function(s) {
        i += s.length + 1;
      }), V[t >> 2] = i, 0;
    }
    function lo(e) {
      return e;
    }
    function Je(e) {
      return e % 4 === 0 && (e % 100 !== 0 || e % 400 === 0);
    }
    function po(e, t) {
      for (var r = 0, i = 0; i <= t; r += e[i++])
        ;
      return r;
    }
    var Yt = [31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31], Jt = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
    function ho(e, t) {
      for (var r = new Date(e.getTime()); t > 0; ) {
        var i = Je(r.getFullYear()), s = r.getMonth(), c = (i ? Yt : Jt)[s];
        if (t > c - r.getDate())
          t -= c - r.getDate() + 1, r.setDate(1), s < 11 ? r.setMonth(s + 1) : (r.setMonth(0), r.setFullYear(r.getFullYear() + 1));
        else
          return r.setDate(r.getDate() + t), r;
      }
      return r;
    }
    function mo(e, t, r) {
      var i = r > 0 ? r : Nt(e) + 1, s = new Array(i), c = Bt(e, s, 0, s.length);
      return t && (s.length = c), s;
    }
    function vo(e, t) {
      I.set(e, t);
    }
    function yo(e, t, r, i) {
      var s = H[i + 40 >> 2], c = { tm_sec: H[i >> 2], tm_min: H[i + 4 >> 2], tm_hour: H[i + 8 >> 2], tm_mday: H[i + 12 >> 2], tm_mon: H[i + 16 >> 2], tm_year: H[i + 20 >> 2], tm_wday: H[i + 24 >> 2], tm_yday: H[i + 28 >> 2], tm_isdst: H[i + 32 >> 2], tm_gmtoff: H[i + 36 >> 2], tm_zone: s ? Ct(s) : "" }, l = Ct(r), f = { "%c": "%a %b %d %H:%M:%S %Y", "%D": "%m/%d/%y", "%F": "%Y-%m-%d", "%h": "%b", "%r": "%I:%M:%S %p", "%R": "%H:%M", "%T": "%H:%M:%S", "%x": "%m/%d/%y", "%X": "%H:%M:%S", "%Ec": "%c", "%EC": "%C", "%Ex": "%m/%d/%y", "%EX": "%H:%M:%S", "%Ey": "%y", "%EY": "%Y", "%Od": "%d", "%Oe": "%e", "%OH": "%H", "%OI": "%I", "%Om": "%m", "%OM": "%M", "%OS": "%S", "%Ou": "%u", "%OU": "%U", "%OV": "%V", "%Ow": "%w", "%OW": "%W", "%Oy": "%y" };
      for (var h in f)
        l = l.replace(new RegExp(h, "g"), f[h]);
      var y = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"], _ = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
      function T(g, P, z) {
        for (var N = typeof g == "number" ? g.toString() : g || ""; N.length < P; )
          N = z[0] + N;
        return N;
      }
      function O(g, P) {
        return T(g, P, "0");
      }
      function x(g, P) {
        function z(ge) {
          return ge < 0 ? -1 : ge > 0 ? 1 : 0;
        }
        var N;
        return (N = z(g.getFullYear() - P.getFullYear())) === 0 && (N = z(g.getMonth() - P.getMonth())) === 0 && (N = z(g.getDate() - P.getDate())), N;
      }
      function W(g) {
        switch (g.getDay()) {
          case 0:
            return new Date(g.getFullYear() - 1, 11, 29);
          case 1:
            return g;
          case 2:
            return new Date(g.getFullYear(), 0, 3);
          case 3:
            return new Date(g.getFullYear(), 0, 2);
          case 4:
            return new Date(g.getFullYear(), 0, 1);
          case 5:
            return new Date(g.getFullYear() - 1, 11, 31);
          case 6:
            return new Date(g.getFullYear() - 1, 11, 30);
        }
      }
      function G(g) {
        var P = ho(new Date(g.tm_year + 1900, 0, 1), g.tm_yday), z = new Date(P.getFullYear(), 0, 4), N = new Date(P.getFullYear() + 1, 0, 4), ge = W(z), Ke = W(N);
        return x(ge, P) <= 0 ? x(Ke, P) <= 0 ? P.getFullYear() + 1 : P.getFullYear() : P.getFullYear() - 1;
      }
      var j = { "%a": function(g) {
        return y[g.tm_wday].substring(0, 3);
      }, "%A": function(g) {
        return y[g.tm_wday];
      }, "%b": function(g) {
        return _[g.tm_mon].substring(0, 3);
      }, "%B": function(g) {
        return _[g.tm_mon];
      }, "%C": function(g) {
        var P = g.tm_year + 1900;
        return O(P / 100 | 0, 2);
      }, "%d": function(g) {
        return O(g.tm_mday, 2);
      }, "%e": function(g) {
        return T(g.tm_mday, 2, " ");
      }, "%g": function(g) {
        return G(g).toString().substring(2);
      }, "%G": function(g) {
        return G(g);
      }, "%H": function(g) {
        return O(g.tm_hour, 2);
      }, "%I": function(g) {
        var P = g.tm_hour;
        return P == 0 ? P = 12 : P > 12 && (P -= 12), O(P, 2);
      }, "%j": function(g) {
        return O(g.tm_mday + po(Je(g.tm_year + 1900) ? Yt : Jt, g.tm_mon - 1), 3);
      }, "%m": function(g) {
        return O(g.tm_mon + 1, 2);
      }, "%M": function(g) {
        return O(g.tm_min, 2);
      }, "%n": function() {
        return `
`;
      }, "%p": function(g) {
        return g.tm_hour >= 0 && g.tm_hour < 12 ? "AM" : "PM";
      }, "%S": function(g) {
        return O(g.tm_sec, 2);
      }, "%t": function() {
        return "	";
      }, "%u": function(g) {
        return g.tm_wday || 7;
      }, "%U": function(g) {
        var P = g.tm_yday + 7 - g.tm_wday;
        return O(Math.floor(P / 7), 2);
      }, "%V": function(g) {
        var P = Math.floor((g.tm_yday + 7 - (g.tm_wday + 6) % 7) / 7);
        if ((g.tm_wday + 371 - g.tm_yday - 2) % 7 <= 2 && P++, P) {
          if (P == 53) {
            var z = (g.tm_wday + 371 - g.tm_yday) % 7;
            z != 4 && (z != 3 || !Je(g.tm_year)) && (P = 1);
          }
        } else {
          P = 52;
          var N = (g.tm_wday + 7 - g.tm_yday - 1) % 7;
          (N == 4 || N == 5 && Je(g.tm_year % 400 - 1)) && P++;
        }
        return O(P, 2);
      }, "%w": function(g) {
        return g.tm_wday;
      }, "%W": function(g) {
        var P = g.tm_yday + 7 - (g.tm_wday + 6) % 7;
        return O(Math.floor(P / 7), 2);
      }, "%y": function(g) {
        return (g.tm_year + 1900).toString().substring(2);
      }, "%Y": function(g) {
        return g.tm_year + 1900;
      }, "%z": function(g) {
        var P = g.tm_gmtoff, z = P >= 0;
        return P = Math.abs(P) / 60, P = P / 60 * 100 + P % 60, (z ? "+" : "-") + ("0000" + P).slice(-4);
      }, "%Z": function(g) {
        return g.tm_zone;
      }, "%%": function() {
        return "%";
      } };
      l = l.replace(/%%/g, "\0\0");
      for (var h in j)
        l.includes(h) && (l = l.replace(new RegExp(h, "g"), j[h](c)));
      l = l.replace(/\0\0/g, "%");
      var q = mo(l, !1);
      return q.length > t ? 0 : (vo(q, e), q.length - 1);
    }
    function go(e, t, r, i, s) {
      return yo(e, t, r, i);
    }
    Rt = o.InternalError = ft(Error, "InternalError"), Zr(), be = o.BindingError = ft(Error, "BindingError"), mn(), sn(), Sn(), Ht = o.UnboundTypeError = ft(Error, "UnboundTypeError"), Rn();
    var Co = { q: Wr, v: jr, a: Hr, h: Vr, l: zr, J: Br, S: Nr, n: Gr, _: qr, d: Lr, ma: Qr, V: Kr, ca: Xr, la: On, ka: En, F: An, ba: xn, R: Un, T: Fn, y: Wn, t: jn, Q: Vn, L: Jn, O: Qn, na: Kn, da: Zn, ha: zt, ia: Xn, ja: to, K: ro, aa: no, $: ao, Y: uo, Z: fo, B: Wo, P: qo, E: jo, p: Uo, b: wo, w: Io, fa: Ho, c: Po, i: Do, j: xo, r: Mo, z: Fo, x: ko, I: zo, N: Bo, D: Lo, H: Yo, W: Qo, U: Ko, k: $o, g: So, e: bo, f: _o, M: Go, m: Ro, o: To, ga: Eo, u: Ao, ea: No, A: Vo, s: Oo, G: Jo, C: lo, X: go };
    Ir();
    var ie = o._free = function() {
      return (ie = o._free = o.asm.qa).apply(null, arguments);
    }, _t = o._malloc = function() {
      return (_t = o._malloc = o.asm.ra).apply(null, arguments);
    }, Qt = function() {
      return (Qt = o.asm.ta).apply(null, arguments);
    };
    o.__embind_initialize_bindings = function() {
      return (o.__embind_initialize_bindings = o.asm.ua).apply(null, arguments);
    };
    var D = function() {
      return (D = o.asm.va).apply(null, arguments);
    }, Me = function() {
      return (Me = o.asm.wa).apply(null, arguments);
    }, M = function() {
      return (M = o.asm.xa).apply(null, arguments);
    }, R = function() {
      return (R = o.asm.ya).apply(null, arguments);
    }, Kt = function() {
      return (Kt = o.asm.za).apply(null, arguments);
    }, Zt = function() {
      return (Zt = o.asm.Aa).apply(null, arguments);
    }, Xt = function() {
      return (Xt = o.asm.Ba).apply(null, arguments);
    }, er = function() {
      return (er = o.asm.Ca).apply(null, arguments);
    };
    o.dynCall_viijii = function() {
      return (o.dynCall_viijii = o.asm.Da).apply(null, arguments);
    };
    var tr = o.dynCall_jiii = function() {
      return (tr = o.dynCall_jiii = o.asm.Ea).apply(null, arguments);
    }, rr = o.dynCall_jiiii = function() {
      return (rr = o.dynCall_jiiii = o.asm.Fa).apply(null, arguments);
    };
    o.dynCall_iiiiij = function() {
      return (o.dynCall_iiiiij = o.asm.Ga).apply(null, arguments);
    }, o.dynCall_iiiiijj = function() {
      return (o.dynCall_iiiiijj = o.asm.Ha).apply(null, arguments);
    }, o.dynCall_iiiiiijj = function() {
      return (o.dynCall_iiiiiijj = o.asm.Ia).apply(null, arguments);
    };
    function wo(e, t) {
      var r = M();
      try {
        return U(e)(t);
      } catch (i) {
        if (R(r), i !== i + 0)
          throw i;
        D(1, 0);
      }
    }
    function _o(e, t, r, i) {
      var s = M();
      try {
        U(e)(t, r, i);
      } catch (c) {
        if (R(s), c !== c + 0)
          throw c;
        D(1, 0);
      }
    }
    function bo(e, t, r) {
      var i = M();
      try {
        U(e)(t, r);
      } catch (s) {
        if (R(i), s !== s + 0)
          throw s;
        D(1, 0);
      }
    }
    function So(e, t) {
      var r = M();
      try {
        U(e)(t);
      } catch (i) {
        if (R(r), i !== i + 0)
          throw i;
        D(1, 0);
      }
    }
    function Po(e, t, r) {
      var i = M();
      try {
        return U(e)(t, r);
      } catch (s) {
        if (R(i), s !== s + 0)
          throw s;
        D(1, 0);
      }
    }
    function $o(e) {
      var t = M();
      try {
        U(e)();
      } catch (r) {
        if (R(t), r !== r + 0)
          throw r;
        D(1, 0);
      }
    }
    function To(e, t, r, i, s, c) {
      var l = M();
      try {
        U(e)(t, r, i, s, c);
      } catch (f) {
        if (R(l), f !== f + 0)
          throw f;
        D(1, 0);
      }
    }
    function Oo(e, t, r, i, s, c, l, f, h, y, _) {
      var T = M();
      try {
        U(e)(t, r, i, s, c, l, f, h, y, _);
      } catch (O) {
        if (R(T), O !== O + 0)
          throw O;
        D(1, 0);
      }
    }
    function Eo(e, t, r, i, s, c, l) {
      var f = M();
      try {
        U(e)(t, r, i, s, c, l);
      } catch (h) {
        if (R(f), h !== h + 0)
          throw h;
        D(1, 0);
      }
    }
    function Ao(e, t, r, i, s, c, l, f) {
      var h = M();
      try {
        U(e)(t, r, i, s, c, l, f);
      } catch (y) {
        if (R(h), y !== y + 0)
          throw y;
        D(1, 0);
      }
    }
    function Do(e, t, r, i) {
      var s = M();
      try {
        return U(e)(t, r, i);
      } catch (c) {
        if (R(s), c !== c + 0)
          throw c;
        D(1, 0);
      }
    }
    function Mo(e, t, r, i, s, c) {
      var l = M();
      try {
        return U(e)(t, r, i, s, c);
      } catch (f) {
        if (R(l), f !== f + 0)
          throw f;
        D(1, 0);
      }
    }
    function Ro(e, t, r, i, s) {
      var c = M();
      try {
        U(e)(t, r, i, s);
      } catch (l) {
        if (R(c), l !== l + 0)
          throw l;
        D(1, 0);
      }
    }
    function xo(e, t, r, i, s) {
      var c = M();
      try {
        return U(e)(t, r, i, s);
      } catch (l) {
        if (R(c), l !== l + 0)
          throw l;
        D(1, 0);
      }
    }
    function ko(e, t, r, i, s, c, l) {
      var f = M();
      try {
        return U(e)(t, r, i, s, c, l);
      } catch (h) {
        if (R(f), h !== h + 0)
          throw h;
        D(1, 0);
      }
    }
    function Uo(e) {
      var t = M();
      try {
        return U(e)();
      } catch (r) {
        if (R(t), r !== r + 0)
          throw r;
        D(1, 0);
      }
    }
    function Fo(e, t, r, i, s, c, l) {
      var f = M();
      try {
        return U(e)(t, r, i, s, c, l);
      } catch (h) {
        if (R(f), h !== h + 0)
          throw h;
        D(1, 0);
      }
    }
    function Io(e, t, r, i) {
      var s = M();
      try {
        return U(e)(t, r, i);
      } catch (c) {
        if (R(s), c !== c + 0)
          throw c;
        D(1, 0);
      }
    }
    function Wo(e, t, r, i) {
      var s = M();
      try {
        return U(e)(t, r, i);
      } catch (c) {
        if (R(s), c !== c + 0)
          throw c;
        D(1, 0);
      }
    }
    function jo(e, t, r, i, s, c) {
      var l = M();
      try {
        return U(e)(t, r, i, s, c);
      } catch (f) {
        if (R(l), f !== f + 0)
          throw f;
        D(1, 0);
      }
    }
    function Lo(e, t, r, i, s, c, l, f, h, y) {
      var _ = M();
      try {
        return U(e)(t, r, i, s, c, l, f, h, y);
      } catch (T) {
        if (R(_), T !== T + 0)
          throw T;
        D(1, 0);
      }
    }
    function Ho(e, t, r) {
      var i = M();
      try {
        return U(e)(t, r);
      } catch (s) {
        if (R(i), s !== s + 0)
          throw s;
        D(1, 0);
      }
    }
    function Vo(e, t, r, i, s, c, l, f, h, y) {
      var _ = M();
      try {
        U(e)(t, r, i, s, c, l, f, h, y);
      } catch (T) {
        if (R(_), T !== T + 0)
          throw T;
        D(1, 0);
      }
    }
    function zo(e, t, r, i, s, c, l, f) {
      var h = M();
      try {
        return U(e)(t, r, i, s, c, l, f);
      } catch (y) {
        if (R(h), y !== y + 0)
          throw y;
        D(1, 0);
      }
    }
    function Bo(e, t, r, i, s, c, l, f, h) {
      var y = M();
      try {
        return U(e)(t, r, i, s, c, l, f, h);
      } catch (_) {
        if (R(y), _ !== _ + 0)
          throw _;
        D(1, 0);
      }
    }
    function No(e, t, r, i, s, c, l, f, h) {
      var y = M();
      try {
        U(e)(t, r, i, s, c, l, f, h);
      } catch (_) {
        if (R(y), _ !== _ + 0)
          throw _;
        D(1, 0);
      }
    }
    function Go(e, t, r, i, s, c, l) {
      var f = M();
      try {
        U(e)(t, r, i, s, c, l);
      } catch (h) {
        if (R(f), h !== h + 0)
          throw h;
        D(1, 0);
      }
    }
    function qo(e, t, r, i) {
      var s = M();
      try {
        return U(e)(t, r, i);
      } catch (c) {
        if (R(s), c !== c + 0)
          throw c;
        D(1, 0);
      }
    }
    function Yo(e, t, r, i, s, c, l, f, h, y, _, T) {
      var O = M();
      try {
        return U(e)(t, r, i, s, c, l, f, h, y, _, T);
      } catch (x) {
        if (R(O), x !== x + 0)
          throw x;
        D(1, 0);
      }
    }
    function Jo(e, t, r, i, s, c, l, f, h, y, _, T, O, x, W, G) {
      var j = M();
      try {
        U(e)(t, r, i, s, c, l, f, h, y, _, T, O, x, W, G);
      } catch (q) {
        if (R(j), q !== q + 0)
          throw q;
        D(1, 0);
      }
    }
    function Qo(e, t, r, i) {
      var s = M();
      try {
        return tr(e, t, r, i);
      } catch (c) {
        if (R(s), c !== c + 0)
          throw c;
        D(1, 0);
      }
    }
    function Ko(e, t, r, i, s) {
      var c = M();
      try {
        return rr(e, t, r, i, s);
      } catch (l) {
        if (R(c), l !== l + 0)
          throw l;
        D(1, 0);
      }
    }
    var Qe;
    he = function e() {
      Qe || nr(), Qe || (he = e);
    };
    function nr() {
      if (te > 0 || (se(), te > 0))
        return;
      function e() {
        Qe || (Qe = !0, o.calledRun = !0, !F && (Z(), u(o), o.onRuntimeInitialized && o.onRuntimeInitialized(), ot()));
      }
      o.setStatus ? (o.setStatus("Running..."), setTimeout(function() {
        setTimeout(function() {
          o.setStatus("");
        }, 1), e();
      }, 1)) : e();
    }
    if (o.preInit)
      for (typeof o.preInit == "function" && (o.preInit = [o.preInit]); o.preInit.length > 0; )
        o.preInit.pop()();
    return nr(), a.ready;
  };
})();
function Si(n) {
  return vr(yr, n);
}
async function Pi(n, {
  tryHarder: a = fe.tryHarder,
  formats: o = fe.formats,
  maxNumberOfSymbols: u = fe.maxNumberOfSymbols
} = fe) {
  return wi(
    n,
    {
      tryHarder: a,
      formats: o,
      maxNumberOfSymbols: u
    },
    yr
  );
}
const $i = [
  "aztec",
  "code_128",
  "code_39",
  "code_93",
  "codabar",
  "data_matrix",
  "ean_13",
  "ean_8",
  "itf",
  "pdf417",
  "qr_code",
  "upc_a",
  "upc_e",
  "unknown"
], gr = /* @__PURE__ */ new Map([
  ["aztec", "Aztec"],
  ["code_128", "Code128"],
  ["code_39", "Code39"],
  ["code_93", "Code93"],
  ["codabar", "Codabar"],
  ["data_matrix", "DataMatrix"],
  ["ean_13", "EAN-13"],
  ["ean_8", "EAN-8"],
  ["itf", "ITF"],
  ["pdf417", "PDF417"],
  ["qr_code", "QRCode"],
  ["upc_a", "UPC-A"],
  ["upc_e", "UPC-E"]
]);
function Ti(n) {
  for (const [a, o] of gr)
    if (n === o)
      return a;
  return "unknown";
}
var et;
class Oi {
  constructor(a = {}) {
    oi(this, et, void 0), Si(), ii(this, et, (a == null ? void 0 : a.formats) ?? []);
  }
  static async getSupportedFormats() {
    return $i;
  }
  async detect(a) {
    const o = await yi(a);
    return o.width === 0 || o.height === 0 ? [] : (await Pi(o, {
      tryHarder: !0,
      formats: ni(this, et).map(
        (u) => gr.get(u)
      )
    })).map((u) => {
      const {
        topLeft: { x: d, y: p },
        topRight: { x: m, y: w },
        bottomLeft: { x: S, y: v },
        bottomRight: { x: b, y: $ }
      } = u.position, C = Math.min(d, m, S, b), E = Math.min(p, w, v, $), A = Math.max(d, m, S, b), F = Math.max(p, w, v, $);
      return {
        boundingBox: new DOMRectReadOnly(C, E, A - C, F - E),
        rawValue: u.text,
        format: Ti(u.format),
        cornerPoints: [
          {
            x: d,
            y: p
          },
          {
            x: m,
            y: w
          },
          {
            x: b,
            y: $
          },
          {
            x: S,
            y: v
          }
        ]
      };
    });
  }
}
et = /* @__PURE__ */ new WeakMap();
globalThis.BarcodeDetector ?? (globalThis.BarcodeDetector = Oi);
const Cr = (n, a, o = "error") => {
  let u, d;
  const p = new Promise(
    (m, w) => {
      u = m, d = w, n.addEventListener(a, u), n.addEventListener(o, d);
    }
  );
  return p.finally(() => {
    n.removeEventListener(a, u), n.removeEventListener(o, d);
  }), p;
}, ur = (n) => new Promise((a) => setTimeout(a, n));
class Ei extends Error {
  constructor() {
    super("can't process cross-origin image"), this.name = "DropImageFetchError";
  }
}
class wr extends Error {
  constructor() {
    super("this browser has no Stream API support"), this.name = "StreamApiNotSupportedError";
  }
}
class Ai extends Error {
  constructor() {
    super(
      "camera access is only permitted in secure context. Use HTTPS or localhost rather than HTTP."
    ), this.name = "InsecureContextError";
  }
}
class Di extends Error {
  constructor() {
    super("Loading camera stream timed out after 3 seconds. If you are on iOS in PWA mode, this is a known issue (see https://github.com/gruhn/vue-qrcode-reader/issues/298)"), this.name = "StreamLoadTimeoutError";
  }
}
const Mi = async (n, {
  detectHandler: a,
  locateHandler: o,
  minDelay: u
}) => {
  const d = new BarcodeDetector({ formats: ["qr_code"] }), p = (m) => async (w) => {
    if (n.readyState > 1) {
      const { lastScanned: S, contentBefore: v, lastScanHadContent: b } = m;
      if (w - S < u)
        window.requestAnimationFrame(p(m));
      else {
        const $ = await d.detect(n), C = $.some((F) => !v.includes(F.rawValue));
        C && a($);
        const E = $.length > 0;
        E && o($), !E && b && o($);
        const A = {
          lastScanned: w,
          lastScanHadContent: E,
          // It can happen that a QR code is constantly in view of the camera but 
          // maybe a scanned frame is a bit blurry and we detect nothing but in the 
          // next frame we detect the code again. We also want to avoid emitting 
          // a `detect` event in such a case. So we don't reset `contentBefore`, 
          // if we detect nothing, only if we detect something new.
          contentBefore: C ? $.map((F) => F.rawValue) : v
        };
        window.requestAnimationFrame(p(A));
      }
    }
  };
  p({
    lastScanned: performance.now(),
    contentBefore: [],
    lastScanHadContent: !1
  })(performance.now());
}, Ri = async (n) => {
  if (n.startsWith("http") && n.includes(location.host) === !1)
    throw new Ei();
  const a = document.createElement("img");
  return a.src = n, await Cr(a, "load"), a;
}, _r = async (n) => await new BarcodeDetector({
  formats: ["qr_code"]
}).detect(n), xi = async (n) => {
  const a = new BarcodeDetector({
    formats: ["qr_code"]
  }), o = await Ri(n);
  return await a.detect(o);
};
var br = {}, J = {};
Object.defineProperty(J, "__esModule", {
  value: !0
});
J.compactObject = $r;
J.deprecated = Vi;
var ki = J.detectBrowser = zi;
J.disableLog = ji;
J.disableWarnings = Li;
J.extractVersion = tt;
J.filterStats = Bi;
J.log = Hi;
J.walkStats = rt;
J.wrapPeerConnectionEvent = Wi;
function Ui(n, a, o) {
  return a = Fi(a), a in n ? Object.defineProperty(n, a, { value: o, enumerable: !0, configurable: !0, writable: !0 }) : n[a] = o, n;
}
function Fi(n) {
  var a = Ii(n, "string");
  return le(a) === "symbol" ? a : String(a);
}
function Ii(n, a) {
  if (le(n) !== "object" || n === null)
    return n;
  var o = n[Symbol.toPrimitive];
  if (o !== void 0) {
    var u = o.call(n, a || "default");
    if (le(u) !== "object")
      return u;
    throw new TypeError("@@toPrimitive must return a primitive value.");
  }
  return (a === "string" ? String : Number)(n);
}
function le(n) {
  "@babel/helpers - typeof";
  return le = typeof Symbol == "function" && typeof Symbol.iterator == "symbol" ? function(a) {
    return typeof a;
  } : function(a) {
    return a && typeof Symbol == "function" && a.constructor === Symbol && a !== Symbol.prototype ? "symbol" : typeof a;
  }, le(n);
}
var Sr = !0, Pr = !0;
function tt(n, a, o) {
  var u = n.match(a);
  return u && u.length >= o && parseInt(u[o], 10);
}
function Wi(n, a, o) {
  if (n.RTCPeerConnection) {
    var u = n.RTCPeerConnection.prototype, d = u.addEventListener;
    u.addEventListener = function(m, w) {
      if (m !== a)
        return d.apply(this, arguments);
      var S = function(b) {
        var $ = o(b);
        $ && (w.handleEvent ? w.handleEvent($) : w($));
      };
      return this._eventMap = this._eventMap || {}, this._eventMap[a] || (this._eventMap[a] = /* @__PURE__ */ new Map()), this._eventMap[a].set(w, S), d.apply(this, [m, S]);
    };
    var p = u.removeEventListener;
    u.removeEventListener = function(m, w) {
      if (m !== a || !this._eventMap || !this._eventMap[a])
        return p.apply(this, arguments);
      if (!this._eventMap[a].has(w))
        return p.apply(this, arguments);
      var S = this._eventMap[a].get(w);
      return this._eventMap[a].delete(w), this._eventMap[a].size === 0 && delete this._eventMap[a], Object.keys(this._eventMap).length === 0 && delete this._eventMap, p.apply(this, [m, S]);
    }, Object.defineProperty(u, "on" + a, {
      get: function() {
        return this["_on" + a];
      },
      set: function(w) {
        this["_on" + a] && (this.removeEventListener(a, this["_on" + a]), delete this["_on" + a]), w && this.addEventListener(a, this["_on" + a] = w);
      },
      enumerable: !0,
      configurable: !0
    });
  }
}
function ji(n) {
  return typeof n != "boolean" ? new Error("Argument type: " + le(n) + ". Please use a boolean.") : (Sr = n, n ? "adapter.js logging disabled" : "adapter.js logging enabled");
}
function Li(n) {
  return typeof n != "boolean" ? new Error("Argument type: " + le(n) + ". Please use a boolean.") : (Pr = !n, "adapter.js deprecation warnings " + (n ? "disabled" : "enabled"));
}
function Hi() {
  if ((typeof window > "u" ? "undefined" : le(window)) === "object") {
    if (Sr)
      return;
    typeof console < "u" && typeof console.log == "function" && console.log.apply(console, arguments);
  }
}
function Vi(n, a) {
  Pr && console.warn(n + " is deprecated, please use " + a + " instead.");
}
function zi(n) {
  var a = {
    browser: null,
    version: null
  };
  if (typeof n > "u" || !n.navigator)
    return a.browser = "Not a browser.", a;
  var o = n.navigator;
  if (o.mozGetUserMedia)
    a.browser = "firefox", a.version = tt(o.userAgent, /Firefox\/(\d+)\./, 1);
  else if (o.webkitGetUserMedia || n.isSecureContext === !1 && n.webkitRTCPeerConnection)
    a.browser = "chrome", a.version = tt(o.userAgent, /Chrom(e|ium)\/(\d+)\./, 2);
  else if (n.RTCPeerConnection && o.userAgent.match(/AppleWebKit\/(\d+)\./))
    a.browser = "safari", a.version = tt(o.userAgent, /AppleWebKit\/(\d+)\./, 1), a.supportsUnifiedPlan = n.RTCRtpTransceiver && "currentDirection" in n.RTCRtpTransceiver.prototype;
  else
    return a.browser = "Not a supported browser.", a;
  return a;
}
function fr(n) {
  return Object.prototype.toString.call(n) === "[object Object]";
}
function $r(n) {
  return fr(n) ? Object.keys(n).reduce(function(a, o) {
    var u = fr(n[o]), d = u ? $r(n[o]) : n[o], p = u && !Object.keys(d).length;
    return d === void 0 || p ? a : Object.assign(a, Ui({}, o, d));
  }, {}) : n;
}
function rt(n, a, o) {
  !a || o.has(a.id) || (o.set(a.id, a), Object.keys(a).forEach(function(u) {
    u.endsWith("Id") ? rt(n, n.get(a[u]), o) : u.endsWith("Ids") && a[u].forEach(function(d) {
      rt(n, n.get(d), o);
    });
  }));
}
function Bi(n, a, o) {
  var u = o ? "outbound-rtp" : "inbound-rtp", d = /* @__PURE__ */ new Map();
  if (a === null)
    return d;
  var p = [];
  return n.forEach(function(m) {
    m.type === "track" && m.trackIdentifier === a.id && p.push(m);
  }), p.forEach(function(m) {
    n.forEach(function(w) {
      w.type === u && w.trackId === m.id && rt(n, w, d);
    });
  }), d;
}
Object.defineProperty(br, "__esModule", {
  value: !0
});
var Ni = br.shimGetUserMedia = Yi, Gi = qi(J);
function Tr(n) {
  if (typeof WeakMap != "function")
    return null;
  var a = /* @__PURE__ */ new WeakMap(), o = /* @__PURE__ */ new WeakMap();
  return (Tr = function(d) {
    return d ? o : a;
  })(n);
}
function qi(n, a) {
  if (!a && n && n.__esModule)
    return n;
  if (n === null || ue(n) !== "object" && typeof n != "function")
    return { default: n };
  var o = Tr(a);
  if (o && o.has(n))
    return o.get(n);
  var u = {}, d = Object.defineProperty && Object.getOwnPropertyDescriptor;
  for (var p in n)
    if (p !== "default" && Object.prototype.hasOwnProperty.call(n, p)) {
      var m = d ? Object.getOwnPropertyDescriptor(n, p) : null;
      m && (m.get || m.set) ? Object.defineProperty(u, p, m) : u[p] = n[p];
    }
  return u.default = n, o && o.set(n, u), u;
}
function ue(n) {
  "@babel/helpers - typeof";
  return ue = typeof Symbol == "function" && typeof Symbol.iterator == "symbol" ? function(a) {
    return typeof a;
  } : function(a) {
    return a && typeof Symbol == "function" && a.constructor === Symbol && a !== Symbol.prototype ? "symbol" : typeof a;
  }, ue(n);
}
var lr = Gi.log;
function Yi(n, a) {
  var o = n && n.navigator;
  if (o.mediaDevices) {
    var u = function(v) {
      if (ue(v) !== "object" || v.mandatory || v.optional)
        return v;
      var b = {};
      return Object.keys(v).forEach(function($) {
        if (!($ === "require" || $ === "advanced" || $ === "mediaSource")) {
          var C = ue(v[$]) === "object" ? v[$] : {
            ideal: v[$]
          };
          C.exact !== void 0 && typeof C.exact == "number" && (C.min = C.max = C.exact);
          var E = function(B, I) {
            return B ? B + I.charAt(0).toUpperCase() + I.slice(1) : I === "deviceId" ? "sourceId" : I;
          };
          if (C.ideal !== void 0) {
            b.optional = b.optional || [];
            var A = {};
            typeof C.ideal == "number" ? (A[E("min", $)] = C.ideal, b.optional.push(A), A = {}, A[E("max", $)] = C.ideal, b.optional.push(A)) : (A[E("", $)] = C.ideal, b.optional.push(A));
          }
          C.exact !== void 0 && typeof C.exact != "number" ? (b.mandatory = b.mandatory || {}, b.mandatory[E("", $)] = C.exact) : ["min", "max"].forEach(function(F) {
            C[F] !== void 0 && (b.mandatory = b.mandatory || {}, b.mandatory[E(F, $)] = C[F]);
          });
        }
      }), v.advanced && (b.optional = (b.optional || []).concat(v.advanced)), b;
    }, d = function(v, b) {
      if (a.version >= 61)
        return b(v);
      if (v = JSON.parse(JSON.stringify(v)), v && ue(v.audio) === "object") {
        var $ = function(B, I, L) {
          I in B && !(L in B) && (B[L] = B[I], delete B[I]);
        };
        v = JSON.parse(JSON.stringify(v)), $(v.audio, "autoGainControl", "googAutoGainControl"), $(v.audio, "noiseSuppression", "googNoiseSuppression"), v.audio = u(v.audio);
      }
      if (v && ue(v.video) === "object") {
        var C = v.video.facingMode;
        C = C && (ue(C) === "object" ? C : {
          ideal: C
        });
        var E = a.version < 66;
        if (C && (C.exact === "user" || C.exact === "environment" || C.ideal === "user" || C.ideal === "environment") && !(o.mediaDevices.getSupportedConstraints && o.mediaDevices.getSupportedConstraints().facingMode && !E)) {
          delete v.video.facingMode;
          var A;
          if (C.exact === "environment" || C.ideal === "environment" ? A = ["back", "rear"] : (C.exact === "user" || C.ideal === "user") && (A = ["front"]), A)
            return o.mediaDevices.enumerateDevices().then(function(F) {
              F = F.filter(function(I) {
                return I.kind === "videoinput";
              });
              var B = F.find(function(I) {
                return A.some(function(L) {
                  return I.label.toLowerCase().includes(L);
                });
              });
              return !B && F.length && A.includes("back") && (B = F[F.length - 1]), B && (v.video.deviceId = C.exact ? {
                exact: B.deviceId
              } : {
                ideal: B.deviceId
              }), v.video = u(v.video), lr("chrome: " + JSON.stringify(v)), b(v);
            });
        }
        v.video = u(v.video);
      }
      return lr("chrome: " + JSON.stringify(v)), b(v);
    }, p = function(v) {
      return a.version >= 64 ? v : {
        name: {
          PermissionDeniedError: "NotAllowedError",
          PermissionDismissedError: "NotAllowedError",
          InvalidStateError: "NotAllowedError",
          DevicesNotFoundError: "NotFoundError",
          ConstraintNotSatisfiedError: "OverconstrainedError",
          TrackStartError: "NotReadableError",
          MediaDeviceFailedDueToShutdown: "NotAllowedError",
          MediaDeviceKillSwitchOn: "NotAllowedError",
          TabCaptureError: "AbortError",
          ScreenCaptureError: "AbortError",
          DeviceCaptureError: "AbortError"
        }[v.name] || v.name,
        message: v.message,
        constraint: v.constraint || v.constraintName,
        toString: function() {
          return this.name + (this.message && ": ") + this.message;
        }
      };
    }, m = function(v, b, $) {
      d(v, function(C) {
        o.webkitGetUserMedia(C, b, function(E) {
          $ && $(p(E));
        });
      });
    };
    if (o.getUserMedia = m.bind(o), o.mediaDevices.getUserMedia) {
      var w = o.mediaDevices.getUserMedia.bind(o.mediaDevices);
      o.mediaDevices.getUserMedia = function(S) {
        return d(S, function(v) {
          return w(v).then(function(b) {
            if (v.audio && !b.getAudioTracks().length || v.video && !b.getVideoTracks().length)
              throw b.getTracks().forEach(function($) {
                $.stop();
              }), new DOMException("", "NotFoundError");
            return b;
          }, function(b) {
            return Promise.reject(p(b));
          });
        });
      };
    }
  }
}
var Or = {};
Object.defineProperty(Or, "__esModule", {
  value: !0
});
var Ji = Or.shimGetUserMedia = Zi, Qi = Ki(J);
function Er(n) {
  if (typeof WeakMap != "function")
    return null;
  var a = /* @__PURE__ */ new WeakMap(), o = /* @__PURE__ */ new WeakMap();
  return (Er = function(d) {
    return d ? o : a;
  })(n);
}
function Ki(n, a) {
  if (!a && n && n.__esModule)
    return n;
  if (n === null || Se(n) !== "object" && typeof n != "function")
    return { default: n };
  var o = Er(a);
  if (o && o.has(n))
    return o.get(n);
  var u = {}, d = Object.defineProperty && Object.getOwnPropertyDescriptor;
  for (var p in n)
    if (p !== "default" && Object.prototype.hasOwnProperty.call(n, p)) {
      var m = d ? Object.getOwnPropertyDescriptor(n, p) : null;
      m && (m.get || m.set) ? Object.defineProperty(u, p, m) : u[p] = n[p];
    }
  return u.default = n, o && o.set(n, u), u;
}
function Se(n) {
  "@babel/helpers - typeof";
  return Se = typeof Symbol == "function" && typeof Symbol.iterator == "symbol" ? function(a) {
    return typeof a;
  } : function(a) {
    return a && typeof Symbol == "function" && a.constructor === Symbol && a !== Symbol.prototype ? "symbol" : typeof a;
  }, Se(n);
}
function Zi(n, a) {
  var o = n && n.navigator, u = n && n.MediaStreamTrack;
  if (o.getUserMedia = function(S, v, b) {
    Qi.deprecated("navigator.getUserMedia", "navigator.mediaDevices.getUserMedia"), o.mediaDevices.getUserMedia(S).then(v, b);
  }, !(a.version > 55 && "autoGainControl" in o.mediaDevices.getSupportedConstraints())) {
    var d = function(v, b, $) {
      b in v && !($ in v) && (v[$] = v[b], delete v[b]);
    }, p = o.mediaDevices.getUserMedia.bind(o.mediaDevices);
    if (o.mediaDevices.getUserMedia = function(S) {
      return Se(S) === "object" && Se(S.audio) === "object" && (S = JSON.parse(JSON.stringify(S)), d(S.audio, "autoGainControl", "mozAutoGainControl"), d(S.audio, "noiseSuppression", "mozNoiseSuppression")), p(S);
    }, u && u.prototype.getSettings) {
      var m = u.prototype.getSettings;
      u.prototype.getSettings = function() {
        var S = m.apply(this, arguments);
        return d(S, "mozAutoGainControl", "autoGainControl"), d(S, "mozNoiseSuppression", "noiseSuppression"), S;
      };
    }
    if (u && u.prototype.applyConstraints) {
      var w = u.prototype.applyConstraints;
      u.prototype.applyConstraints = function(S) {
        return this.kind === "audio" && Se(S) === "object" && (S = JSON.parse(JSON.stringify(S)), d(S, "autoGainControl", "mozAutoGainControl"), d(S, "noiseSuppression", "mozNoiseSuppression")), w.apply(this, [S]);
      };
    }
  }
}
var X = {};
Object.defineProperty(X, "__esModule", {
  value: !0
});
X.shimAudioContext = ca;
X.shimCallbacksAPI = na;
X.shimConstraints = Mr;
X.shimCreateOfferLegacy = sa;
var Xi = X.shimGetUserMedia = oa;
X.shimLocalStreamsAPI = ta;
X.shimRTCIceServerUrls = ia;
X.shimRemoteStreamsAPI = ra;
X.shimTrackEventTransceiver = aa;
var Ar = ea(J);
function Dr(n) {
  if (typeof WeakMap != "function")
    return null;
  var a = /* @__PURE__ */ new WeakMap(), o = /* @__PURE__ */ new WeakMap();
  return (Dr = function(d) {
    return d ? o : a;
  })(n);
}
function ea(n, a) {
  if (!a && n && n.__esModule)
    return n;
  if (n === null || de(n) !== "object" && typeof n != "function")
    return { default: n };
  var o = Dr(a);
  if (o && o.has(n))
    return o.get(n);
  var u = {}, d = Object.defineProperty && Object.getOwnPropertyDescriptor;
  for (var p in n)
    if (p !== "default" && Object.prototype.hasOwnProperty.call(n, p)) {
      var m = d ? Object.getOwnPropertyDescriptor(n, p) : null;
      m && (m.get || m.set) ? Object.defineProperty(u, p, m) : u[p] = n[p];
    }
  return u.default = n, o && o.set(n, u), u;
}
function de(n) {
  "@babel/helpers - typeof";
  return de = typeof Symbol == "function" && typeof Symbol.iterator == "symbol" ? function(a) {
    return typeof a;
  } : function(a) {
    return a && typeof Symbol == "function" && a.constructor === Symbol && a !== Symbol.prototype ? "symbol" : typeof a;
  }, de(n);
}
function ta(n) {
  if (!(de(n) !== "object" || !n.RTCPeerConnection)) {
    if ("getLocalStreams" in n.RTCPeerConnection.prototype || (n.RTCPeerConnection.prototype.getLocalStreams = function() {
      return this._localStreams || (this._localStreams = []), this._localStreams;
    }), !("addStream" in n.RTCPeerConnection.prototype)) {
      var a = n.RTCPeerConnection.prototype.addTrack;
      n.RTCPeerConnection.prototype.addStream = function(u) {
        var d = this;
        this._localStreams || (this._localStreams = []), this._localStreams.includes(u) || this._localStreams.push(u), u.getAudioTracks().forEach(function(p) {
          return a.call(d, p, u);
        }), u.getVideoTracks().forEach(function(p) {
          return a.call(d, p, u);
        });
      }, n.RTCPeerConnection.prototype.addTrack = function(u) {
        for (var d = this, p = arguments.length, m = new Array(p > 1 ? p - 1 : 0), w = 1; w < p; w++)
          m[w - 1] = arguments[w];
        return m && m.forEach(function(S) {
          d._localStreams ? d._localStreams.includes(S) || d._localStreams.push(S) : d._localStreams = [S];
        }), a.apply(this, arguments);
      };
    }
    "removeStream" in n.RTCPeerConnection.prototype || (n.RTCPeerConnection.prototype.removeStream = function(u) {
      var d = this;
      this._localStreams || (this._localStreams = []);
      var p = this._localStreams.indexOf(u);
      if (p !== -1) {
        this._localStreams.splice(p, 1);
        var m = u.getTracks();
        this.getSenders().forEach(function(w) {
          m.includes(w.track) && d.removeTrack(w);
        });
      }
    });
  }
}
function ra(n) {
  if (!(de(n) !== "object" || !n.RTCPeerConnection) && ("getRemoteStreams" in n.RTCPeerConnection.prototype || (n.RTCPeerConnection.prototype.getRemoteStreams = function() {
    return this._remoteStreams ? this._remoteStreams : [];
  }), !("onaddstream" in n.RTCPeerConnection.prototype))) {
    Object.defineProperty(n.RTCPeerConnection.prototype, "onaddstream", {
      get: function() {
        return this._onaddstream;
      },
      set: function(u) {
        var d = this;
        this._onaddstream && (this.removeEventListener("addstream", this._onaddstream), this.removeEventListener("track", this._onaddstreampoly)), this.addEventListener("addstream", this._onaddstream = u), this.addEventListener("track", this._onaddstreampoly = function(p) {
          p.streams.forEach(function(m) {
            if (d._remoteStreams || (d._remoteStreams = []), !d._remoteStreams.includes(m)) {
              d._remoteStreams.push(m);
              var w = new Event("addstream");
              w.stream = m, d.dispatchEvent(w);
            }
          });
        });
      }
    });
    var a = n.RTCPeerConnection.prototype.setRemoteDescription;
    n.RTCPeerConnection.prototype.setRemoteDescription = function() {
      var u = this;
      return this._onaddstreampoly || this.addEventListener("track", this._onaddstreampoly = function(d) {
        d.streams.forEach(function(p) {
          if (u._remoteStreams || (u._remoteStreams = []), !(u._remoteStreams.indexOf(p) >= 0)) {
            u._remoteStreams.push(p);
            var m = new Event("addstream");
            m.stream = p, u.dispatchEvent(m);
          }
        });
      }), a.apply(u, arguments);
    };
  }
}
function na(n) {
  if (!(de(n) !== "object" || !n.RTCPeerConnection)) {
    var a = n.RTCPeerConnection.prototype, o = a.createOffer, u = a.createAnswer, d = a.setLocalDescription, p = a.setRemoteDescription, m = a.addIceCandidate;
    a.createOffer = function(v, b) {
      var $ = arguments.length >= 2 ? arguments[2] : arguments[0], C = o.apply(this, [$]);
      return b ? (C.then(v, b), Promise.resolve()) : C;
    }, a.createAnswer = function(v, b) {
      var $ = arguments.length >= 2 ? arguments[2] : arguments[0], C = u.apply(this, [$]);
      return b ? (C.then(v, b), Promise.resolve()) : C;
    };
    var w = function(v, b, $) {
      var C = d.apply(this, [v]);
      return $ ? (C.then(b, $), Promise.resolve()) : C;
    };
    a.setLocalDescription = w, w = function(v, b, $) {
      var C = p.apply(this, [v]);
      return $ ? (C.then(b, $), Promise.resolve()) : C;
    }, a.setRemoteDescription = w, w = function(v, b, $) {
      var C = m.apply(this, [v]);
      return $ ? (C.then(b, $), Promise.resolve()) : C;
    }, a.addIceCandidate = w;
  }
}
function oa(n) {
  var a = n && n.navigator;
  if (a.mediaDevices && a.mediaDevices.getUserMedia) {
    var o = a.mediaDevices, u = o.getUserMedia.bind(o);
    a.mediaDevices.getUserMedia = function(d) {
      return u(Mr(d));
    };
  }
  !a.getUserMedia && a.mediaDevices && a.mediaDevices.getUserMedia && (a.getUserMedia = function(p, m, w) {
    a.mediaDevices.getUserMedia(p).then(m, w);
  }.bind(a));
}
function Mr(n) {
  return n && n.video !== void 0 ? Object.assign({}, n, {
    video: Ar.compactObject(n.video)
  }) : n;
}
function ia(n) {
  if (n.RTCPeerConnection) {
    var a = n.RTCPeerConnection;
    n.RTCPeerConnection = function(u, d) {
      if (u && u.iceServers) {
        for (var p = [], m = 0; m < u.iceServers.length; m++) {
          var w = u.iceServers[m];
          w.urls === void 0 && w.url ? (Ar.deprecated("RTCIceServer.url", "RTCIceServer.urls"), w = JSON.parse(JSON.stringify(w)), w.urls = w.url, delete w.url, p.push(w)) : p.push(u.iceServers[m]);
        }
        u.iceServers = p;
      }
      return new a(u, d);
    }, n.RTCPeerConnection.prototype = a.prototype, "generateCertificate" in a && Object.defineProperty(n.RTCPeerConnection, "generateCertificate", {
      get: function() {
        return a.generateCertificate;
      }
    });
  }
}
function aa(n) {
  de(n) === "object" && n.RTCTrackEvent && "receiver" in n.RTCTrackEvent.prototype && !("transceiver" in n.RTCTrackEvent.prototype) && Object.defineProperty(n.RTCTrackEvent.prototype, "transceiver", {
    get: function() {
      return {
        receiver: this.receiver
      };
    }
  });
}
function sa(n) {
  var a = n.RTCPeerConnection.prototype.createOffer;
  n.RTCPeerConnection.prototype.createOffer = function(u) {
    if (u) {
      typeof u.offerToReceiveAudio < "u" && (u.offerToReceiveAudio = !!u.offerToReceiveAudio);
      var d = this.getTransceivers().find(function(m) {
        return m.receiver.track.kind === "audio";
      });
      u.offerToReceiveAudio === !1 && d ? d.direction === "sendrecv" ? d.setDirection ? d.setDirection("sendonly") : d.direction = "sendonly" : d.direction === "recvonly" && (d.setDirection ? d.setDirection("inactive") : d.direction = "inactive") : u.offerToReceiveAudio === !0 && !d && this.addTransceiver("audio", {
        direction: "recvonly"
      }), typeof u.offerToReceiveVideo < "u" && (u.offerToReceiveVideo = !!u.offerToReceiveVideo);
      var p = this.getTransceivers().find(function(m) {
        return m.receiver.track.kind === "video";
      });
      u.offerToReceiveVideo === !1 && p ? p.direction === "sendrecv" ? p.setDirection ? p.setDirection("sendonly") : p.direction = "sendonly" : p.direction === "recvonly" && (p.setDirection ? p.setDirection("inactive") : p.direction = "inactive") : u.offerToReceiveVideo === !0 && !p && this.addTransceiver("video", {
        direction: "recvonly"
      });
    }
    return a.apply(this, arguments);
  };
}
function ca(n) {
  de(n) !== "object" || n.AudioContext || (n.AudioContext = n.webkitAudioContext);
}
const ua = (n) => {
  let a = !1, o;
  return (...u) => (a || (o = n(u), a = !0), o);
}, fa = ua(() => {
  const n = ki(window);
  switch (n.browser) {
    case "chrome":
      Ni(window, n);
      break;
    case "firefox":
      Ji(window, n);
      break;
    case "safari":
      Xi(window, n);
      break;
    default:
      throw new wr();
  }
});
let ae = { isActive: !1 };
function bt() {
  if (ae.isActive) {
    ae.videoEl.src = "", ae.videoEl.srcObject = null, ae.videoEl.load();
    for (const n of ae.stream.getTracks())
      ae.stream.removeTrack(n), n.stop();
    ae = { isActive: !1 };
  }
}
async function la(n, { constraints: a, torch: o }) {
  var p, m;
  if (window.isSecureContext !== !0)
    throw new Ai();
  if (((p = navigator == null ? void 0 : navigator.mediaDevices) == null ? void 0 : p.getUserMedia) === void 0)
    throw new wr();
  fa();
  const u = await navigator.mediaDevices.getUserMedia({
    audio: !1,
    video: a
  });
  if (n.srcObject !== void 0 ? n.srcObject = u : n.mozSrcObject !== void 0 ? n.mozSrcObject = u : window.URL.createObjectURL ? n.src = window.URL.createObjectURL(u) : window.webkitURL ? n.src = window.webkitURL.createObjectURL(u) : n.src = u.id, await Promise.race([
    Cr(n, "loadeddata"),
    // On iOS devices in PWA mode, QrcodeStream works initially, but after
    // killing and restarting the PWA, all video elements fail to load camera 
    // streams and never emit the `loadeddata` event. Looks like this is 
    // related to a WebKit issue (see #298). No workarounds at the moment. 
    // To at least detect this situation, we throw an error if the event
    // has not been emitted after a 3 second timeout.
    ur(3e3).then(() => {
      throw new Di();
    })
  ]), await ur(500), o) {
    const [w] = u.getVideoTracks();
    w.getCapabilities().torch ? w.applyConstraints({ advanced: [{ torch: !0 }] }) : console.warn("device does not support torch capability");
  }
  ae = { videoEl: n, stream: u, isActive: !0 };
  const [d] = ae.stream.getVideoTracks();
  return ((m = d == null ? void 0 : d.getCapabilities) == null ? void 0 : m.call(d)) ?? {};
}
const da = { class: "qrcode-stream-wrapper" }, pa = { class: "qrcode-stream-overlay" }, ha = /* @__PURE__ */ St({
  __name: "QrcodeStream",
  props: {
    constraints: {
      type: Object,
      default() {
        return { facingMode: "environment" };
      }
    },
    paused: {
      type: Boolean,
      default: !1
    },
    torch: {
      type: Boolean,
      default: !1
    },
    track: {
      type: Function
    }
  },
  emits: ["detect", "camera-on", "camera-off", "error"],
  setup(n, { emit: a }) {
    const o = n, u = Re(), d = Re(), p = Re(), m = Re(!1), w = Re(!1);
    Zo(() => {
      w.value = !0;
    }), Xo(() => {
      bt();
    });
    const S = or(() => ({
      torch: o.torch,
      constraints: o.constraints,
      shouldStream: w.value && !o.paused
    }));
    ir(S, async (C) => {
      if (C.shouldStream)
        try {
          const E = await la(p.value, C);
          w.value ? (m.value = !0, a("camera-on", E)) : bt();
        } catch (E) {
          a("error", E);
        }
      else {
        const E = u.value, A = E.getContext("2d"), F = p.value;
        E.width = F.videoWidth, E.height = F.videoHeight, A == null || A.drawImage(F, 0, 0, F.videoWidth, F.videoHeight), bt(), m.value = !1, a("camera-off");
      }
    }, { deep: !0 });
    const v = or(() => S.value.shouldStream && m.value);
    ir(v, (C) => {
      if (C) {
        b(u.value), b(d.value);
        const E = () => o.track === void 0 ? 500 : 40;
        Mi(p.value, {
          detectHandler: (A) => a("detect", A),
          locateHandler: $,
          minDelay: E()
        });
      }
    });
    const b = (C) => {
      const E = C.getContext("2d");
      console.assert(
        E !== void 0,
        "tried to clear canvas with undefined 2D context"
      ), E.clearRect(0, 0, C.width, C.height);
    }, $ = (C) => {
      const E = d.value, A = p.value;
      if (console.assert(
        E !== void 0 && A !== void 0,
        "onLocate handler called although component is not mounted"
      ), C.length === 0 || o.track === void 0)
        b(E);
      else {
        const F = A.offsetWidth, B = A.offsetHeight, I = A.videoWidth, L = A.videoHeight, ee = Math.max(F / I, B / L), pe = I * ee, H = L * ee, V = pe / I, ke = H / L, Ue = (F - pe) / 2, Fe = (B - H) / 2, Ce = ({ x: se, y: Z }) => ({
          x: Math.floor(se * V),
          y: Math.floor(Z * ke)
        }), Pe = ({ x: se, y: Z }) => ({
          x: Math.floor(se + Ue),
          y: Math.floor(Z + Fe)
        }), Ie = C.map((se) => {
          const { boundingBox: Z, cornerPoints: ot } = se, { x: it, y: at } = Pe(
            Ce({
              x: Z.x,
              y: Z.y
            })
          ), { x: st, y: te } = Ce({
            x: Z.width,
            y: Z.height
          });
          return {
            ...se,
            cornerPoints: ot.map((he) => Pe(Ce(he))),
            boundingBox: DOMRectReadOnly.fromRect({ x: it, y: at, width: st, height: te })
          };
        });
        E.width = A.offsetWidth, E.height = A.offsetHeight;
        const We = E.getContext("2d");
        o.track(Ie, We);
      }
    };
    return (C, E) => (Pt(), $t("div", da, [
      Ze("video", {
        ref_key: "videoRef",
        ref: p,
        class: ei([{ "qrcode-stream-camera--hidden": !v.value }, "qrcode-stream-camera"]),
        autoplay: "",
        muted: "",
        playsinline: ""
      }, null, 2),
      ti(Ze("canvas", {
        ref_key: "pauseFrameRef",
        ref: u,
        class: "qrcode-stream-camera"
      }, null, 512), [
        [ri, !v.value]
      ]),
      Ze("canvas", {
        ref_key: "trackingLayerRef",
        ref: d,
        class: "qrcode-stream-overlay"
      }, null, 512),
      Ze("div", pa, [
        dr(C.$slots, "default", {}, void 0, !0)
      ])
    ]));
  }
});
const ma = (n, a) => {
  const o = n.__vccOpts || n;
  for (const [u, d] of a)
    o[u] = d;
  return o;
}, va = /* @__PURE__ */ ma(ha, [["__scopeId", "data-v-44e7ec8a"]]), ya = /* @__PURE__ */ St({
  __name: "QrcodeCapture",
  emits: ["detect"],
  setup(n, { emit: a }) {
    const o = async (d) => {
      const p = await await d;
      a("detect", p);
    }, u = (d) => {
      if (!(d.target instanceof HTMLInputElement) || !d.target.files)
        return;
      const p = [...Array.from(d.target.files)];
      for (const m of p.map(_r))
        o(m);
    };
    return (d, p) => (Pt(), $t("input", {
      onChange: u,
      type: "file",
      name: "image",
      accept: "image/*",
      capture: "environment",
      multiple: ""
    }, null, 32));
  }
}), ga = ["onDrop"], Ca = /* @__PURE__ */ St({
  __name: "QrcodeDropZone",
  emits: ["detect", "dragover", "error"],
  setup(n, { emit: a }) {
    const o = async (p) => {
      try {
        const m = await p;
        a("detect", m);
      } catch (m) {
        a("error", m);
      }
    }, u = (p) => {
      a("dragover", p);
    }, d = ({ dataTransfer: p }) => {
      if (!p)
        return;
      u(!1);
      const m = [...Array.from(p.files)], w = p.getData("text/uri-list");
      m.forEach((S) => {
        o(_r(S));
      }), w !== "" && o(xi(w));
    };
    return (p, m) => (Pt(), $t("div", {
      onDrop: Xe(d, ["prevent", "stop"]),
      onDragenter: m[0] || (m[0] = Xe((w) => u(!0), ["prevent", "stop"])),
      onDragleave: m[1] || (m[1] = Xe((w) => u(!1), ["prevent", "stop"])),
      onDragover: m[2] || (m[2] = Xe(() => {
      }, ["prevent", "stop"]))
    }, [
      dr(p.$slots, "default")
    ], 40, ga));
  }
});
function wa(n) {
  n.component("qrcode-stream", va), n.component("qrcode-capture", ya), n.component("qrcode-drop-zone", Ca);
}
const _a = { install: wa };
let nt = null;
typeof window < "u" ? nt = window.Vue : typeof global < "u" && (nt = global.Vue);
nt && nt.use(_a);
export {
  ya as QrcodeCapture,
  Ca as QrcodeDropZone,
  va as QrcodeStream,
  _a as VueQrcodeReader,
  wa as install
};
